<!doctype html><html lang=en class=no-js> <head><script type=application/ld+json>
    {
      "@context": "https://schema.org",
      "@id": "https://hds-sandbox.github.io/RDM_NGS_course/",
      "@type": "LearningResource",
      "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
      "description": "This workshop a basic intro to RDM, and how to apply good practices to your NGS data",
      "keywords": [
        "RDM, Research Data Management, NGS, Bioinformatics, Data Management Plan"
      ],
      "learningResourceType": [
        "e-learning"
      ],
      "license": [
        {
          "@id": "# Link to license webpage (https://creativecommons.org/licenses/by-sa/4.0/)",
          "@type": "CreativeWork"
        }
      ],
      "mentions": [
        {
          "@id": "# Link to DOI or zenodo materials (10.5281/zenodo.7660225)",
          "@type": "Thing"
        },
        {
          "@id": "#Another link to DOI or zenodo materials (10.5281/zenodo.7565997)",
          "@type": "Thing"
        }
      ],
      "name": "Introduction to NGS Data Management",
      "educationalLevel": "Beginner",
      "audience": "PhD, postdocs, MSc, etc"
    }
    </script><title>Next Generation Sequencing Data - RDM for NGS course</title><link rel=stylesheet href=assets/stylesheets/main.fad675c6.min.css><link rel=stylesheet href=assets/stylesheets/palette.356b1318.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=stylesheets/neoteroi-mkdocs.css><link rel=stylesheet href=stylesheets/extra.css><script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script id=__analytics>function __md_analytics(){function n(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],n("js",new Date),n("config",""),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&n("event","search",{search_term:this.value})}),document$.subscribe(function(){var a=document.forms.feedback;if(void 0!==a)for(var e of a.querySelectorAll("[type=submit]"))e.addEventListener("click",function(e){e.preventDefault();var t=document.location.pathname,e=this.getAttribute("data-md-value");n("event","feedback",{page:t,data:e}),a.firstElementChild.disabled=!0;e=a.querySelector(".md-feedback__note [data-md-value='"+e+"']");e&&(e.hidden=!1)}),a.hidden=!1}),location$.subscribe(function(e){n("config","",{page_path:e.pathname})})});var e=document.createElement("script");e.async=!0,e.src="https://www.googletagmanager.com/gtag/js?id=",document.getElementById("__analytics").insertAdjacentElement("afterEnd",e)}</script><script>"undefined"!=typeof __md_analytics&&__md_analytics()</script></head> <body dir=ltr data-md-color-scheme=brightness data-md-color-primary=indigo data-md-color-accent=indigo> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#next-generation-sequencing-data class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class="md-header md-header--shadow md-header--lifted" data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=http://hds-sandbox.github.io/ title="RDM for NGS course" class="md-header__button md-logo" aria-label="RDM for NGS course" data-md-component=logo> <img src=images/logo.png alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> RDM for NGS course </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Next Generation Sequencing Data </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=brightness data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" title=Share aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg> </a> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/hds-sandbox/RDM_NGS_course title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> RDM_NGS_course </div> </a> </div> </nav> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=index.html class=md-tabs__link> Introduction </a> </li> <li class="md-tabs__item md-tabs__item--active"> <a href=01_RDM_intro.html class=md-tabs__link> Course contents </a> </li> <li class=md-tabs__item> <a href=practical_workshop.html class=md-tabs__link> Practical workshop </a> </li> <li class=md-tabs__item> <a href=keywords.html class=md-tabs__link> Keyword index </a> </li> <li class=md-tabs__item> <a href=contributors.html class=md-tabs__link> Contributors </a> </li> </ul> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=http://hds-sandbox.github.io/ title="RDM for NGS course" class="md-nav__button md-logo" aria-label="RDM for NGS course" data-md-component=logo> <img src=images/logo.png alt=logo> </a> RDM for NGS course </label> <div class=md-nav__source> <a href=https://github.com/hds-sandbox/RDM_NGS_course title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> RDM_NGS_course </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=index.html class=md-nav__link> <span class=md-ellipsis> Introduction </span> </a> </li> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2 checked> <label class=md-nav__link for=__nav_2 id=__nav_2_label tabindex> <span class=md-ellipsis> Course contents </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_2_label aria-expanded=true> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Course contents </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=01_RDM_intro.html class=md-nav__link> <span class=md-ellipsis> What is RDM </span> </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> <span class=md-ellipsis> NGS data </span> <span class="md-nav__icon md-icon"></span> </label> <a href=02_NGS_data.html class="md-nav__link md-nav__link--active"> <span class=md-ellipsis> NGS data </span> </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#what-is-next-generation-sequencing class=md-nav__link> <span class=md-ellipsis> What is Next Generation Sequencing </span> </a> </li> <li class=md-nav__item> <a href=#data-produced-during-an-ngs-experiment class=md-nav__link> <span class=md-ellipsis> Data produced during an NGS experiment </span> </a> <nav class=md-nav aria-label="Data produced during an NGS experiment"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#electronic-laboratory-notebook class=md-nav__link> <span class=md-ellipsis> Electronic Laboratory Notebook </span> </a> </li> <li class=md-nav__item> <a href=#laboratory-protocols class=md-nav__link> <span class=md-ellipsis> Laboratory protocols </span> </a> </li> <li class=md-nav__item> <a href=#samples class=md-nav__link> <span class=md-ellipsis> Samples </span> </a> </li> <li class=md-nav__item> <a href=#library-preparation class=md-nav__link> <span class=md-ellipsis> Library Preparation </span> </a> </li> <li class=md-nav__item> <a href=#sequencing class=md-nav__link> <span class=md-ellipsis> Sequencing </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#data-produced-during-preprocessing class=md-nav__link> <span class=md-ellipsis> Data produced during preprocessing </span> </a> <nav class=md-nav aria-label="Data produced during preprocessing"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#quality-control class=md-nav__link> <span class=md-ellipsis> Quality Control </span> </a> </li> <li class=md-nav__item> <a href=#data-alignment class=md-nav__link> <span class=md-ellipsis> Data Alignment </span> </a> </li> <li class=md-nav__item> <a href=#variant-calling class=md-nav__link> <span class=md-ellipsis> Variant Calling </span> </a> </li> <li class=md-nav__item> <a href=#rna-expression-analysis class=md-nav__link> <span class=md-ellipsis> RNA Expression Analysis </span> </a> </li> <li class=md-nav__item> <a href=#epigenetic-profiling class=md-nav__link> <span class=md-ellipsis> Epigenetic Profiling </span> </a> </li> <li class=md-nav__item> <a href=#metagenomic-analysis class=md-nav__link> <span class=md-ellipsis> Metagenomic Analysis </span> </a> </li> <li class=md-nav__item> <a href=#other-types-of-data class=md-nav__link> <span class=md-ellipsis> Other types of data </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#data-produced-during-data-analysis class=md-nav__link> <span class=md-ellipsis> Data produced during data analysis </span> </a> <nav class=md-nav aria-label="Data produced during data analysis"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#variant-annotation class=md-nav__link> <span class=md-ellipsis> Variant Annotation </span> </a> </li> <li class=md-nav__item> <a href=#differential-expression-analysis class=md-nav__link> <span class=md-ellipsis> Differential Expression Analysis </span> </a> </li> <li class=md-nav__item> <a href=#differential-binding-or-chromatin-accesibility-analysis class=md-nav__link> <span class=md-ellipsis> Differential Binding or Chromatin Accesibility Analysis </span> </a> </li> <li class=md-nav__item> <a href=#functional-analysis class=md-nav__link> <span class=md-ellipsis> Functional analysis </span> </a> </li> <li class=md-nav__item> <a href=#metagenomic-analysis_1 class=md-nav__link> <span class=md-ellipsis> Metagenomic Analysis </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#other-types-of-data_1 class=md-nav__link> <span class=md-ellipsis> Other types of data </span> </a> </li> <li class=md-nav__item> <a href=#other-resources-used class=md-nav__link> <span class=md-ellipsis> Other resources used </span> </a> <nav class=md-nav aria-label="Other resources used"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#knowledge-databases class=md-nav__link> <span class=md-ellipsis> Knowledge databases </span> </a> </li> <li class=md-nav__item> <a href=#code-notebooks-and-scripts class=md-nav__link> <span class=md-ellipsis> Code notebooks and scripts </span> </a> </li> <li class=md-nav__item> <a href=#workflows-and-pipelines class=md-nav__link> <span class=md-ellipsis> Workflows and pipelines </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#data-formats-summary class=md-nav__link> <span class=md-ellipsis> Data formats summary </span> </a> </li> <li class=md-nav__item> <a href=#wrap-up class=md-nav__link> <span class=md-ellipsis> Wrap up </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=03_data_life_cycle.html class=md-nav__link> <span class=md-ellipsis> Data Life Cycle </span> </a> </li> <li class=md-nav__item> <a href=04_OS_FAIR.html class=md-nav__link> <span class=md-ellipsis> Open Science and FAIR principles </span> </a> </li> <li class=md-nav__item> <a href=05_DMP.html class=md-nav__link> <span class=md-ellipsis> Data Management Plans </span> </a> </li> <li class=md-nav__item> <a href=06_file_structure.html class=md-nav__link> <span class=md-ellipsis> File structure and naming conventions </span> </a> </li> <li class=md-nav__item> <a href=07_metadata.html class=md-nav__link> <span class=md-ellipsis> Metadata applied to NGS data </span> </a> </li> <li class=md-nav__item> <a href=08_database.html class=md-nav__link> <span class=md-ellipsis> Database of your data and projects </span> </a> </li> <li class=md-nav__item> <a href=09_version_control.html class=md-nav__link> <span class=md-ellipsis> Version control of your data analysis </span> </a> </li> <li class=md-nav__item> <a href=10_repos.html class=md-nav__link> <span class=md-ellipsis> Archiving your data </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=practical_workshop.html class=md-nav__link> <span class=md-ellipsis> Practical workshop </span> </a> </li> <li class=md-nav__item> <a href=keywords.html class=md-nav__link> <span class=md-ellipsis> Keyword index </span> </a> </li> <li class=md-nav__item> <a href=contributors.html class=md-nav__link> <span class=md-ellipsis> Contributors </span> </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#what-is-next-generation-sequencing class=md-nav__link> <span class=md-ellipsis> What is Next Generation Sequencing </span> </a> </li> <li class=md-nav__item> <a href=#data-produced-during-an-ngs-experiment class=md-nav__link> <span class=md-ellipsis> Data produced during an NGS experiment </span> </a> <nav class=md-nav aria-label="Data produced during an NGS experiment"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#electronic-laboratory-notebook class=md-nav__link> <span class=md-ellipsis> Electronic Laboratory Notebook </span> </a> </li> <li class=md-nav__item> <a href=#laboratory-protocols class=md-nav__link> <span class=md-ellipsis> Laboratory protocols </span> </a> </li> <li class=md-nav__item> <a href=#samples class=md-nav__link> <span class=md-ellipsis> Samples </span> </a> </li> <li class=md-nav__item> <a href=#library-preparation class=md-nav__link> <span class=md-ellipsis> Library Preparation </span> </a> </li> <li class=md-nav__item> <a href=#sequencing class=md-nav__link> <span class=md-ellipsis> Sequencing </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#data-produced-during-preprocessing class=md-nav__link> <span class=md-ellipsis> Data produced during preprocessing </span> </a> <nav class=md-nav aria-label="Data produced during preprocessing"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#quality-control class=md-nav__link> <span class=md-ellipsis> Quality Control </span> </a> </li> <li class=md-nav__item> <a href=#data-alignment class=md-nav__link> <span class=md-ellipsis> Data Alignment </span> </a> </li> <li class=md-nav__item> <a href=#variant-calling class=md-nav__link> <span class=md-ellipsis> Variant Calling </span> </a> </li> <li class=md-nav__item> <a href=#rna-expression-analysis class=md-nav__link> <span class=md-ellipsis> RNA Expression Analysis </span> </a> </li> <li class=md-nav__item> <a href=#epigenetic-profiling class=md-nav__link> <span class=md-ellipsis> Epigenetic Profiling </span> </a> </li> <li class=md-nav__item> <a href=#metagenomic-analysis class=md-nav__link> <span class=md-ellipsis> Metagenomic Analysis </span> </a> </li> <li class=md-nav__item> <a href=#other-types-of-data class=md-nav__link> <span class=md-ellipsis> Other types of data </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#data-produced-during-data-analysis class=md-nav__link> <span class=md-ellipsis> Data produced during data analysis </span> </a> <nav class=md-nav aria-label="Data produced during data analysis"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#variant-annotation class=md-nav__link> <span class=md-ellipsis> Variant Annotation </span> </a> </li> <li class=md-nav__item> <a href=#differential-expression-analysis class=md-nav__link> <span class=md-ellipsis> Differential Expression Analysis </span> </a> </li> <li class=md-nav__item> <a href=#differential-binding-or-chromatin-accesibility-analysis class=md-nav__link> <span class=md-ellipsis> Differential Binding or Chromatin Accesibility Analysis </span> </a> </li> <li class=md-nav__item> <a href=#functional-analysis class=md-nav__link> <span class=md-ellipsis> Functional analysis </span> </a> </li> <li class=md-nav__item> <a href=#metagenomic-analysis_1 class=md-nav__link> <span class=md-ellipsis> Metagenomic Analysis </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#other-types-of-data_1 class=md-nav__link> <span class=md-ellipsis> Other types of data </span> </a> </li> <li class=md-nav__item> <a href=#other-resources-used class=md-nav__link> <span class=md-ellipsis> Other resources used </span> </a> <nav class=md-nav aria-label="Other resources used"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#knowledge-databases class=md-nav__link> <span class=md-ellipsis> Knowledge databases </span> </a> </li> <li class=md-nav__item> <a href=#code-notebooks-and-scripts class=md-nav__link> <span class=md-ellipsis> Code notebooks and scripts </span> </a> </li> <li class=md-nav__item> <a href=#workflows-and-pipelines class=md-nav__link> <span class=md-ellipsis> Workflows and pipelines </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#data-formats-summary class=md-nav__link> <span class=md-ellipsis> Data formats summary </span> </a> </li> <li class=md-nav__item> <a href=#wrap-up class=md-nav__link> <span class=md-ellipsis> Wrap up </span> </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <nav class=md-tags> <a href=keywords.html#ngs class=md-tag>NGS</a> <a href=keywords.html#bioinformatics-data class=md-tag>Bioinformatics data</a> <a href=keywords.html#data-formats class=md-tag>Data formats</a> </nav> <div><h1 id=next-generation-sequencing-data>Next Generation Sequencing data<a class=headerlink href=#next-generation-sequencing-data title="Permanent link">Â¶</a></h1> <p><strong>Last updated:</strong> <em><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">November 30, 2023</span></em></p> <div class="admonition note"> <p class=admonition-title>Section Overview</p> <p>ðŸ•° <strong>Time Estimation:</strong> X minutes </p> <p>ðŸ’¬ <strong>Learning Objectives:</strong> </p> <ol> <li>Learn about what is Next Generation Sequencing </li> <li>Learn about different types of NGS data</li> <li>Learn about the specific challenges regarding RDM for NGS data</li> </ol> </div> <p>In this session, we'll explore the essentials of Next Generation Sequencing technology, which generates vast data volumes in a single experiment. We'll examine the types of data produced during an NGS experiment and the results created from the downstream analysis. Moreover, we'll address specific challenges NGS encounters in RDM, such as, managing genomic resources, pipelines/workflows, data analysis, data preservation and documentation. We will also mention briefly concerns regarding sensitive data and GDPR protection of NGS data.</p> <h2 id=what-is-next-generation-sequencing>What is Next Generation Sequencing<a class=headerlink href=#what-is-next-generation-sequencing title="Permanent link">Â¶</a></h2> <p>Next Generation Sequencing (NGS), also known as high-throughput sequencing, is a revolutionary technology that has transformed genomics research in recent years. NGS is a suite of advanced DNA sequencing techniques that enable rapid and cost-effective analysis of DNA or RNA molecules. Unlike traditional Sanger sequencing, which could only sequence a limited number of DNA fragments at a time, NGS can analyze millions of DNA fragments simultaneously in a single run. This massive parallel sequencing capacity has drastically increased the speed, efficiency, and scale of DNA sequencing, making it an indispensable tool in modern genomics and biomedical studies.</p> <p>NGS workflows encompass key steps, starting with sample preparation, where DNA or RNA is extracted and fragmented into smaller segments. Subsequently, unique identifiers are added to the fragments during library preparation, enabling multiplexed sequencing. The fragments are then amplified and sequenced in parallel using state-of-the-art NGS platforms. Finally, data analysis methods are applied to process the raw sequencing data, reconstruct the original DNA or RNA sequence, and identify genetic variations, structural changes, or functional elements.</p> <p>The versatility of NGS extends far beyond genome sequencing. It is widely utilized in various applications, including transcriptome analysis (RNA-Seq), epigenetic profiling (ChIP-Seq), metagenomics, and targeted sequencing. NGS has revolutionized fields like oncology, infectious disease research, and personalized medicine, as its ability to generate vast amounts of data rapidly provides unprecedented insights into the genetic basis of diseases and biological processes. As NGS technologies continue to advance and become more accessible, they will remain at the front of cutting-edge genomics research, driving innovations that contribute to our understanding of complex genetic interactions and their implications for human health and biology.</p> <h2 id=data-produced-during-an-ngs-experiment>Data produced during an NGS experiment<a class=headerlink href=#data-produced-during-an-ngs-experiment title="Permanent link">Â¶</a></h2> <p>During an NGS experiment, various types of data are generated, including everything needed to create and prepare the samples used for sequencing. These compass, between others, laboratory protocols, lab notebooks, NGS libraries and the sequencing data itself.</p> <h3 id=electronic-laboratory-notebook>Electronic Laboratory Notebook<a class=headerlink href=#electronic-laboratory-notebook title="Permanent link">Â¶</a></h3> <p>An electronic laboratory notebook (ELN) is a digital version of the traditional paper notebook used by scientists, researchers, and professionals in laboratories and research settings. It serves as a platform for recording, organizing, and managing experimental data, observations, protocols, and other research-related information in electronic format. ELNs offer a range of features including data entry, text editing, file attachments, collaboration tools, and data search capabilities. They provide advantages over paper notebooks by enabling easy data retrieval, backup, and sharing, as well as facilitating collaboration among researchers.</p> <h3 id=laboratory-protocols>Laboratory protocols<a class=headerlink href=#laboratory-protocols title="Permanent link">Â¶</a></h3> <p>A laboratory protocol is documented set of procedures and methodologies outlining the precise steps required to prepare and manage samples of an experiment.</p> <h3 id=samples>Samples<a class=headerlink href=#samples title="Permanent link">Â¶</a></h3> <p>In the context of molecular biology and Next-Generation Sequencing (NGS), a sample refers to a specific biological material or substance that is collected and prepared for genetic analysis. This material can encompass various biological entities such as DNA, RNA, or proteins, depending on the objectives of the study. For instance, a DNA sample might be extracted from a tissue or blood sample to analyze genetic variations or sequences. In NGS experiments, samples are typically processed according to specific protocols to generate large amounts of sequencing data, allowing researchers to delve into the genetic information contained within the sample.</p> <div class="admonition note"> <p class=admonition-title>Bioinformatics data from your samples</p> <p>The experimental conditions of each of the samples should be registered in a metadata file that can be used later in your data analyses, such as treatment, cell type, timepoints, etc.</p> </div> <h3 id=library-preparation>Library Preparation<a class=headerlink href=#library-preparation title="Permanent link">Â¶</a></h3> <p>After preparing your samples, the next step in an NGS experiment is the library preparation, where DNA or RNA samples are processed to create DNA fragments with specific adapter sequences. These adapters contain unique barcode sequences that allow the identification of individual samples within a multiplexed sequencing run.</p> <h3 id=sequencing>Sequencing<a class=headerlink href=#sequencing title="Permanent link">Â¶</a></h3> <p>Once the libraries are prepared, they undergo the sequencing process using NGS platforms such as Illumina, Oxford Nanopore Technologies (ONT), or Pacific Biosciences (PacBio). The choice of platform depends on factors such as read length, error rate, throughput, and cost. The sequencing generates millions to billions of short DNA or RNA fragments, known as reads.</p> <p>The sequencing process produces raw data in the form of short reads, each representing a small segment of the genome or transcriptome. These reads are typically stored in FASTQ files, which contain nucleotide sequences and corresponding quality scores for each read.</p> <div class="admonition note"> <p class=admonition-title>Bioinformatics data from sequencing</p> <ul> <li><strong>FASTQ:</strong> This is one of the primary file formats used to store raw NGS data. FASTQ files contain nucleotide sequences and their corresponding quality scores. Each read from the sequencing process is represented as a set of four lines: a read identifier, the sequence of nucleotides, a separator (usually a '+'), and the quality scores for each base in the sequence.</li> </ul> </div> <h2 id=data-produced-during-preprocessing>Data produced during preprocessing<a class=headerlink href=#data-produced-during-preprocessing title="Permanent link">Â¶</a></h2> <h3 id=quality-control>Quality Control<a class=headerlink href=#quality-control title="Permanent link">Â¶</a></h3> <p>Before proceeding with data analysis, quality control is performed to assess the accuracy and reliability of the raw sequencing data. Low-quality reads and potential sequencing artifacts are filtered out to ensure the reliability of downstream analyses. Most useful tools to track Quality Control are FastQC and MultiQC.</p> <p>!!! "Bioinformatics data from QC"</p> <div class=highlight><pre><span></span><code>- **FastQC**: FastQC is a widely-used bioinformatics tool that provides a comprehensive assessment of the quality of sequencing data, generating valuable insights into potential issues or biases in the data. 
- **MultiQC**: MultiQC is a companion tool that simplifies the process of aggregating and visualizing quality control metrics from multiple samples or datasets, allowing for a quick overview of data quality across an entire project.
</code></pre></div> <h3 id=data-alignment>Data Alignment<a class=headerlink href=#data-alignment title="Permanent link">Â¶</a></h3> <p>In DNA sequencing, the reads are aligned to a reference genome using alignment algorithms to determine the specific location of each read in the genome. For RNA-Seq experiments, reads are aligned to a reference transcriptome to identify gene expression levels.</p> <p>!!! "Bioinformatics data from alignment"</p> <div class=highlight><pre><span></span><code>- **BAM/SAM**: These are binary and text versions, respectively, of Sequence Alignment/Map (SAM) files. SAM files store the alignment information of sequencing reads to a reference genome or transcriptome. BAM files are compressed and more space-efficient, making them the preferred format for storing aligned reads.
- **FASTA**: FASTA files store nucleotide or amino acid sequences, and they are often used for reference sequences or assembled contigs. Each sequence in a FASTA file begins with a single-line description, followed by the actual sequence data.
- **GTF/GFF**: These files are used to annotate genomic features, such as genes, exons, and transcripts. Gene Transfer Format (GTF) and General Feature Format (GFF) files include information on feature positions, names, and additional attributes.
- **Alignment indexes**: These are reference data structures that facilitate efficient and rapid mapping of sequencing reads to a reference genome or transcriptome during Next Generation Sequencing (NGS) data analysis.
</code></pre></div> <h3 id=variant-calling>Variant Calling<a class=headerlink href=#variant-calling title="Permanent link">Â¶</a></h3> <p>For genomic DNA sequencing, variant calling identifies differences (mutations, insertions, deletions) between the sequenced sample and the reference genome. This step is crucial for detecting genetic variations associated with diseases or phenotypic traits.</p> <p>!!! "Bioinformatics data from VC"</p> <div class=highlight><pre><span></span><code>- **VCF**: Variant Call Format (VCF) files store genetic variations, such as single nucleotide variants (SNVs), insertions, deletions, and structural variants, identified during variant calling. VCF files include variant position, alleles, genotype information, and quality scores.
</code></pre></div> <h3 id=rna-expression-analysis>RNA Expression Analysis<a class=headerlink href=#rna-expression-analysis title="Permanent link">Â¶</a></h3> <p>In RNA-Seq experiments, data analysis includes quantification of gene expression levels, detection of alternative splicing events, and identification of differentially expressed genes under different experimental conditions.</p> <p>!!! "Bioinformatics data from Expression Analyses"</p> <div class=highlight><pre><span></span><code>- **Count matrix**: represents the quantified number of times each gene or genomic feature is observed in a set of biological samples. Each row typically corresponds to a gene, while each column represents a sample.
</code></pre></div> <h3 id=epigenetic-profiling>Epigenetic Profiling<a class=headerlink href=#epigenetic-profiling title="Permanent link">Â¶</a></h3> <p>Epigenetic analyses, such as ChIP-Seq, assess DNA modifications and protein-DNA interactions, providing insights into gene regulation and chromatin structure.</p> <p>!!! "Bioinformatics data from Epigenetic profiling"</p> <div class=highlight><pre><span></span><code>- **Peak Calling Results**: For ChIP-Seq or other genomic profiling experiments, peak calling identifies regions with enriched signal intensity. The results are typically presented in BED or BEDGraph formats, indicating the genomic positions and signal intensities of detected peaks, or BigWig files for visualization in a genome browser.
- **BED/BEDGraph**: These files are used to represent genomic intervals, such as gene coordinates, regions of interest, or coverage information. BED files define genomic regions with start and end coordinates, while BEDGraph files represent continuous data (e.g., coverage) as a graph.
- **WIG/BigWig**: These files store genome-wide data, such as coverage, signal intensity, or ChIP-Seq peaks. Wiggle (WIG) files contain continuous data, while BigWig files are binary, compressed versions that enable efficient random access to large datasets.
</code></pre></div> <h3 id=metagenomic-analysis>Metagenomic Analysis<a class=headerlink href=#metagenomic-analysis title="Permanent link">Â¶</a></h3> <p>For metagenomics, the NGS data is used to characterize microbial communities in environmental samples, including the identification of species and functional genes. Overall, the data generated during an NGS experiment is extensive and diverse, providing a wealth of information that is crucial for a wide range of biological and medical research applications. The interpretation and analysis of this data require sophisticated bioinformatics tools and pipelines, as well as domain-specific knowledge to derive meaningful biological insights.</p> <h3 id=other-types-of-data>Other types of data<a class=headerlink href=#other-types-of-data title="Permanent link">Â¶</a></h3> <p>There are many more data formats we have not explored here. You are welcome to check more bioinformatics data formats at the <a href=http://genome.ucsc.edu/FAQ/FAQformat>UCSC webpage</a>.</p> <p><img alt=bioinfo_data_format src=images/UCSC_data_FAQ.png></p> <div class="admonition tip"> <p class=admonition-title>Bioinformatics data format tutorial</p> <p>Check out <a href=https://bioinformatics.uconn.edu/resources-and-events/tutorials-2/file-formats-tutorial/ >this tutorial</a> for a more in-depth explanation of different bioinformatics data formats</p> </div> <h2 id=data-produced-during-data-analysis>Data produced during data analysis<a class=headerlink href=#data-produced-during-data-analysis title="Permanent link">Â¶</a></h2> <p>Results from NGS data analysis often involve a variety of visualizations and tabular outputs that provide valuable insights into the biological significance of the data. Here are some common types of results that researchers may obtain from NGS data analysis.</p> <h3 id=variant-annotation>Variant Annotation<a class=headerlink href=#variant-annotation title="Permanent link">Â¶</a></h3> <p>After variant calling, the identified genetic variations are annotated with functional information, such as their impact on genes, regulatory elements, or protein products.</p> <h3 id=differential-expression-analysis>Differential Expression Analysis<a class=headerlink href=#differential-expression-analysis title="Permanent link">Â¶</a></h3> <p>Differential Expression Analysis Tables: Differential expression analysis compares gene expression levels between different experimental conditions or groups. The results are typically presented in tabular format, listing genes with significant changes in expression, along with log-fold changes and statistical significance values.</p> <h3 id=differential-binding-or-chromatin-accesibility-analysis>Differential Binding or Chromatin Accesibility Analysis<a class=headerlink href=#differential-binding-or-chromatin-accesibility-analysis title="Permanent link">Â¶</a></h3> <p>Differential Binding Analysis Tables: Differential expression analysis compares chromatin accessibility or binding levels between different experimental conditions or groups. The results are typically presented in tabular format, listing genes with significant changes in binging or chromatin accesibility, along with log-fold changes and statistical significance values.</p> <h3 id=functional-analysis>Functional analysis<a class=headerlink href=#functional-analysis title="Permanent link">Â¶</a></h3> <p>Functional analysis identifies functional categories, pathways, or gene sets that are significantly overrepresented in a list of differentially expressed genes. The results are presented as tables with enrichment scores and adjusted p-values, indicating the biological relevance of gene sets.</p> <h3 id=metagenomic-analysis_1>Metagenomic Analysis<a class=headerlink href=#metagenomic-analysis_1 title="Permanent link">Â¶</a></h3> <p>Metagenomic Taxonomic Abundance Tables: In metagenomics, taxonomic abundance tables show the relative abundance of different microbial taxa across multiple samples. They are essential for understanding the composition of microbial communities.</p> <p>Metagenomic Functional Analysis Results: Metagenomic functional analysis identifies the functional capabilities of microbial communities. Results are typically presented as tables showing the relative abundance of functional genes or pathways.</p> <h2 id=other-types-of-data_1>Other types of data<a class=headerlink href=#other-types-of-data_1 title="Permanent link">Â¶</a></h2> <p>These results from NGS data analysis play a crucial role in interpreting the biological significance of the data, enabling researchers to draw meaningful conclusions and generate hypotheses for further investigation. Effective data visualization and clear tabular outputs are essential for effectively communicating and interpreting the wealth of information generated by NGS experiments.</p> <div class="admonition note"> <p class=admonition-title>Bioinformatics visualizations</p> <ul> <li><strong>Heatmaps</strong>: Heatmaps are graphical representations of data matrices, where each cell's color intensity reflects the value of a specific parameter. In NGS data analysis, heatmaps are frequently used to visualize gene expression patterns, epigenetic modifications, or microbial abundances across different samples or conditions.</li> <li><strong>Volcano Plots</strong>: Volcano plots display the relationship between fold change (usually log-fold change) and statistical significance (e.g., p-value) for each gene in a differential analysis. Genes with high significance and large fold changes are often represented further away from the plot's center.</li> <li><strong>Genome Browser Snapshots</strong>: Genome browser snapshots display aligned sequencing reads and various genomic features (e.g., gene annotations, ChIP-Seq peaks) in a genomic region of interest. These snapshots provide a visual representation of NGS data aligned to a reference genome.</li> <li><strong>Network Visualizations</strong>: Network visualizations depict the interactions between genes, proteins, or other biological entities, providing insights into complex biological networks. These visualizations can help uncover regulatory relationships or functional modules.</li> <li><strong>Genomic Annotations</strong>: Results may include annotations for genetic variations, such as their functional impact on genes, genomic regions, or regulatory elements.</li> </ul> </div> <h2 id=other-resources-used>Other resources used<a class=headerlink href=#other-resources-used title="Permanent link">Â¶</a></h2> <h3 id=knowledge-databases>Knowledge databases<a class=headerlink href=#knowledge-databases title="Permanent link">Â¶</a></h3> <p>A knowledge database is a structured repository of biological information that categorizes and annotates genes, proteins, and their functions, facilitating comprehensive understanding and analysis of biological systems. Here are five examples of knowledge databases:</p> <ol> <li><strong>Gene Ontology (GO)</strong>: A comprehensive resource that classifies gene functions into defined terms, allowing for standardized annotation and comparison of genes across different organisms.</li> <li><strong>Disease Ontology</strong>: A database that provides structured, standardized terminology for various diseases and their relationships, aiding in the systematic analysis of disease-related data.</li> <li><strong>KEGG Pathways</strong>: A collection of manually curated pathway maps representing molecular interactions and reaction networks within cells, enabling the interpretation of high-throughput data in the context of biological systems.</li> <li><strong>Reactome</strong>: An open-access database that offers curated descriptions of biological processes, including pathways, reactions, and molecular events, facilitating the interpretation of large-scale biological data.</li> <li><strong>UniProt</strong>: An extensive protein knowledgebase that provides detailed information about proteins, including their sequences, functions, and related annotations, supporting a wide range of biological research endeavors.</li> </ol> <h3 id=code-notebooks-and-scripts>Code notebooks and scripts<a class=headerlink href=#code-notebooks-and-scripts title="Permanent link">Â¶</a></h3> <p>Code notebooks and scripts are essential tools used to analyze Next Generation Sequencing (NGS) data, providing researchers with a documented and reproducible workflow for data processing, and analyses. Common programming languages (and notebooks) are: Python (Jupyter Notebooks), R (Rmarkdown), perl and bash.</p> <h3 id=workflows-and-pipelines>Workflows and pipelines<a class=headerlink href=#workflows-and-pipelines title="Permanent link">Â¶</a></h3> <p>These are structured sets of interconnected data analysis steps and bioinformatics tools that streamline and automate the transformation of raw sequencing reads into biologically meaningful information, ensuring reproducibility and efficiency in NGS data analysis. Some examples of workflow and pipeline languages are:</p> <ul> <li><strong>Nextflow</strong>: Nextflow is a popular workflow management system that enables scalable and portable NGS data analysis pipelines, allowing researchers to process data across various computing environments and platforms.</li> <li><strong>Snakemake</strong>: Snakemake is a workflow management system that uses Python-based scripting to create flexible and automated NGS data analysis pipelines, facilitating parallel processing and easy integration with existing tools.</li> </ul> <p>A great example of community curated workflows is the nf-core community. Nf-core is a collaborative and open-source initiative comprising bioinformaticians and researchers dedicated to developing and maintaining a collection of curated and reproducible Nextflow-based pipelines for NGS data analysis, ensuring standardized and efficient data processing workflows.</p> <h2 id=data-formats-summary>Data formats summary<a class=headerlink href=#data-formats-summary title="Permanent link">Â¶</a></h2> <p>Here is a summary of different file formats used for NGS data and its results. It's important to select appropriate file formats that balance data accessibility, storage efficiency, and compatibility with downstream analysis tools. Additionally, the use of standardized file formats facilitates data sharing and collaboration among researchers in the scientific community.</p> <ul> <li>Tabular formats: File formats like CSV, TSV, and XLSX used to store data in rows and columns for easy data analysis and sharing.</li> <li>Image formats: File formats such as PNG and SVG used to store graphical visualizations, making them easily viewable and shareable.</li> <li>Binary formats: File formats like NPZ and H5 used to store large datasets, ensuring efficient data access and storage.</li> <li>BED: A format for storing genomic interval data, often used for representing genomic features and peak calling results.</li> <li>GTF: A format for storing genomic feature annotations, such as genes and transcripts, providing information on their positions and attributes.</li> <li>VCF: Variant Call Format used to store genetic variations and their attributes identified during variant calling.</li> <li>JSON: A lightweight data-interchange format for storing hierarchical data structures, commonly used in bioinformatics tools.</li> <li>HTML: A format used to create interactive reports that include both visualizations and textual descriptions of analysis results.</li> <li>FASTQ: A format for storing raw NGS data, containing nucleotide sequences and quality scores for each read.</li> <li>FASTA: A format for storing nucleotide or amino acid sequences, often used for reference sequences or assembled contigs.</li> <li>BAM/SAM: BAM is A binary format used to store compressed alignment data, representing aligned reads to a reference genome. SAM is a text version of the BAM format, representing aligned reads to a reference genome.</li> <li>Alignment indexes: Reference data structures facilitating efficient mapping of reads to a reference genome during NGS data analysis.</li> <li>FASTQC: A tool generating quality control reports for raw NGS data, assessing read quality and detecting potential issues.</li> <li>MultiQC: A tool aggregating results from multiple QC analyses, simplifying the overview and comparison of multiple datasets.</li> <li>Code notebooks: Interactive documents combining code, visualizations, and explanatory text, aiding in data analysis reproducibility and documentation.</li> <li>Scripts: Text files containing sets of commands or code instructions for automating data processing and analysis tasks.</li> </ul> <h2 id=wrap-up>Wrap up<a class=headerlink href=#wrap-up title="Permanent link">Â¶</a></h2> <p>In this lesson we have taken a look a the vast and diverse landscape of bioinformatics data. This should give you a start in understanding the type of data we will be focusing on for the next lessons. In the next chapter we will talk about its <a href=03_data_life_cycle.html>life cycle</a>, which will give you an idea about what do you need to pay attention in order to organize, document and share your data.</p></div> <hr> <div class=md-source-file> <small> Last update: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">November 30, 2023</span> <br> Created: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">November 30, 2023</span> </small> </div> </article> </div> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </button> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=01_RDM_intro.html class="md-footer__link md-footer__link--prev" aria-label="Previous: What is RDM"> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <span class=md-footer__direction> Previous </span> <div class=md-ellipsis> What is RDM </div> </div> </a> <a href=03_data_life_cycle.html class="md-footer__link md-footer__link--next" aria-label="Next: Data Life Cycle"> <div class=md-footer__title> <span class=md-footer__direction> Next </span> <div class=md-ellipsis> Data Life Cycle </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> CC-BY-SA 4.0 license </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=https://github.com/hds-sandbox target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://twitter.com/ucph_heads target=_blank rel=noopener title=twitter.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": ".", "features": ["content.code.annotate", "content.tooltips", "navigation.indexes", "navigation.sections", "navigation.footer", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow"], "search": "assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script> <script src=assets/javascripts/bundle.cd18aaf1.min.js></script> <script src=javascripts/mathjax.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>