<!doctype html><html lang=en class=no-js> <head><script type=application/ld+json>
    {
      "@context": "https://schema.org",
      "@id": "https://hds-sandbox.github.io/RDM_NGS_course/",
      "@type": "LearningResource",
      "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
      "description": "This workshop a basic intro to RDM, and how to apply good practices to your NGS data",
      "keywords": [
        "RDM, Research Data Management, NGS, Bioinformatics, Data Management Plan"
      ],
      "learningResourceType": [
        "e-learning"
      ],
      "license": [
        {
          "@id": "# Link to license webpage (https://creativecommons.org/licenses/by-sa/4.0/)",
          "@type": "CreativeWork"
        }
      ],
      "mentions": [
        {
          "@id": "# Link to DOI or zenodo materials (10.5281/zenodo.7660225)",
          "@type": "Thing"
        },
        {
          "@id": "#Another link to DOI or zenodo materials (10.5281/zenodo.7565997)",
          "@type": "Thing"
        }
      ],
      "name": "Introduction to NGS Data Management",
      "educationalLevel": "Beginner",
      "audience": "PhD, postdocs, MSc, etc"
    }
    </script><title>Metadata for NGS data - RDM for NGS course</title><link rel=stylesheet href=assets/stylesheets/main.fad675c6.min.css><link rel=stylesheet href=assets/stylesheets/palette.356b1318.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=stylesheets/neoteroi-mkdocs.css><link rel=stylesheet href=stylesheets/extra.css><script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script id=__analytics>function __md_analytics(){function n(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],n("js",new Date),n("config",""),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&n("event","search",{search_term:this.value})}),document$.subscribe(function(){var a=document.forms.feedback;if(void 0!==a)for(var e of a.querySelectorAll("[type=submit]"))e.addEventListener("click",function(e){e.preventDefault();var t=document.location.pathname,e=this.getAttribute("data-md-value");n("event","feedback",{page:t,data:e}),a.firstElementChild.disabled=!0;e=a.querySelector(".md-feedback__note [data-md-value='"+e+"']");e&&(e.hidden=!1)}),a.hidden=!1}),location$.subscribe(function(e){n("config","",{page_path:e.pathname})})});var e=document.createElement("script");e.async=!0,e.src="https://www.googletagmanager.com/gtag/js?id=",document.getElementById("__analytics").insertAdjacentElement("afterEnd",e)}</script><script>"undefined"!=typeof __md_analytics&&__md_analytics()</script></head> <body dir=ltr data-md-color-scheme=brightness data-md-color-primary=indigo data-md-color-accent=indigo> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#metadata-for-ngs-data class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class="md-header md-header--shadow md-header--lifted" data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=http://hds-sandbox.github.io/ title="RDM for NGS course" class="md-header__button md-logo" aria-label="RDM for NGS course" data-md-component=logo> <img src=images/logo.png alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> RDM for NGS course </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Metadata for NGS data </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=brightness data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" title=Share aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg> </a> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/hds-sandbox/RDM_NGS_course title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> RDM_NGS_course </div> </a> </div> </nav> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=index.html class=md-tabs__link> Introduction </a> </li> <li class="md-tabs__item md-tabs__item--active"> <a href=01_RDM_intro.html class=md-tabs__link> Course contents </a> </li> <li class=md-tabs__item> <a href=practical_workshop.html class=md-tabs__link> Practical workshop </a> </li> <li class=md-tabs__item> <a href=keywords.html class=md-tabs__link> Keyword index </a> </li> <li class=md-tabs__item> <a href=contributors.html class=md-tabs__link> Contributors </a> </li> </ul> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=http://hds-sandbox.github.io/ title="RDM for NGS course" class="md-nav__button md-logo" aria-label="RDM for NGS course" data-md-component=logo> <img src=images/logo.png alt=logo> </a> RDM for NGS course </label> <div class=md-nav__source> <a href=https://github.com/hds-sandbox/RDM_NGS_course title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> RDM_NGS_course </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=index.html class=md-nav__link> <span class=md-ellipsis> Introduction </span> </a> </li> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2 checked> <label class=md-nav__link for=__nav_2 id=__nav_2_label tabindex> <span class=md-ellipsis> Course contents </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_2_label aria-expanded=true> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Course contents </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=01_RDM_intro.html class=md-nav__link> <span class=md-ellipsis> What is RDM </span> </a> </li> <li class=md-nav__item> <a href=02_NGS_data.html class=md-nav__link> <span class=md-ellipsis> NGS data </span> </a> </li> <li class=md-nav__item> <a href=03_data_life_cycle.html class=md-nav__link> <span class=md-ellipsis> Data Life Cycle </span> </a> </li> <li class=md-nav__item> <a href=04_OS_FAIR.html class=md-nav__link> <span class=md-ellipsis> Open Science and FAIR principles </span> </a> </li> <li class=md-nav__item> <a href=05_DMP.html class=md-nav__link> <span class=md-ellipsis> Data Management Plans </span> </a> </li> <li class=md-nav__item> <a href=06_file_structure.html class=md-nav__link> <span class=md-ellipsis> File structure and naming conventions </span> </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> <span class=md-ellipsis> Metadata applied to NGS data </span> <span class="md-nav__icon md-icon"></span> </label> <a href=07_metadata.html class="md-nav__link md-nav__link--active"> <span class=md-ellipsis> Metadata applied to NGS data </span> </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#what-is-metadata-and-why-it-is-important class=md-nav__link> <span class=md-ellipsis> What is metadata and why it is important </span> </a> </li> <li class=md-nav__item> <a href=#how-to-collect-metadata-in-your-folders class=md-nav__link> <span class=md-ellipsis> How to collect metadata in your folders </span> </a> <nav class=md-nav aria-label="How to collect metadata in your folders"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#readmemd-file class=md-nav__link> <span class=md-ellipsis> README.md file </span> </a> </li> <li class=md-nav__item> <a href=#metadatayml class=md-nav__link> <span class=md-ellipsis> metadata.yml </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#metadata-fields class=md-nav__link> <span class=md-ellipsis> Metadata fields </span> </a> <nav class=md-nav aria-label="Metadata fields"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#general-metadata-fields class=md-nav__link> <span class=md-ellipsis> General metadata fields </span> </a> </li> <li class=md-nav__item> <a href=#assay-metadata-fields class=md-nav__link> <span class=md-ellipsis> Assay metadata fields </span> </a> </li> <li class=md-nav__item> <a href=#sample-metadata-fields class=md-nav__link> <span class=md-ellipsis> Sample metadata fields </span> </a> </li> <li class=md-nav__item> <a href=#project-metadata-fields class=md-nav__link> <span class=md-ellipsis> Project metadata fields </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#more-info class=md-nav__link> <span class=md-ellipsis> More info </span> </a> </li> <li class=md-nav__item> <a href=#wrap-up class=md-nav__link> <span class=md-ellipsis> Wrap up </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=08_database.html class=md-nav__link> <span class=md-ellipsis> Database of your data and projects </span> </a> </li> <li class=md-nav__item> <a href=09_version_control.html class=md-nav__link> <span class=md-ellipsis> Version control of your data analysis </span> </a> </li> <li class=md-nav__item> <a href=10_repos.html class=md-nav__link> <span class=md-ellipsis> Archiving your data </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=practical_workshop.html class=md-nav__link> <span class=md-ellipsis> Practical workshop </span> </a> </li> <li class=md-nav__item> <a href=keywords.html class=md-nav__link> <span class=md-ellipsis> Keyword index </span> </a> </li> <li class=md-nav__item> <a href=contributors.html class=md-nav__link> <span class=md-ellipsis> Contributors </span> </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#what-is-metadata-and-why-it-is-important class=md-nav__link> <span class=md-ellipsis> What is metadata and why it is important </span> </a> </li> <li class=md-nav__item> <a href=#how-to-collect-metadata-in-your-folders class=md-nav__link> <span class=md-ellipsis> How to collect metadata in your folders </span> </a> <nav class=md-nav aria-label="How to collect metadata in your folders"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#readmemd-file class=md-nav__link> <span class=md-ellipsis> README.md file </span> </a> </li> <li class=md-nav__item> <a href=#metadatayml class=md-nav__link> <span class=md-ellipsis> metadata.yml </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#metadata-fields class=md-nav__link> <span class=md-ellipsis> Metadata fields </span> </a> <nav class=md-nav aria-label="Metadata fields"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#general-metadata-fields class=md-nav__link> <span class=md-ellipsis> General metadata fields </span> </a> </li> <li class=md-nav__item> <a href=#assay-metadata-fields class=md-nav__link> <span class=md-ellipsis> Assay metadata fields </span> </a> </li> <li class=md-nav__item> <a href=#sample-metadata-fields class=md-nav__link> <span class=md-ellipsis> Sample metadata fields </span> </a> </li> <li class=md-nav__item> <a href=#project-metadata-fields class=md-nav__link> <span class=md-ellipsis> Project metadata fields </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#more-info class=md-nav__link> <span class=md-ellipsis> More info </span> </a> </li> <li class=md-nav__item> <a href=#wrap-up class=md-nav__link> <span class=md-ellipsis> Wrap up </span> </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <div><h1 id=metadata-for-ngs-data>Metadata for NGS data<a class=headerlink href=#metadata-for-ngs-data title="Permanent link">Â¶</a></h1> <p><strong>Last updated:</strong> <em><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">November 30, 2023</span></em></p> <div class="admonition note"> <p class=admonition-title>Section Overview</p> <p>ðŸ•° <strong>Time Estimation:</strong> X minutes </p> <p>ðŸ’¬ <strong>Learning Objectives:</strong> </p> <ol> <li>Learn what is metadata </li> <li>Learn what metadata is associated to NGS data</li> <li>Learn sources for controlled vocabularies for NGS data</li> </ol> </div> <p>Metadata is the behind-the-scenes information that makes sense of data. It's the extra layer of details that help you understand what the data is all about. Imagine you have a book â€“ the words on the pages are the data, while the title, author, and table of contents are the metadata. In a similar way, metadata gives data context and structure. For NGS data, metadata includes information such as when and where the data was collected, what it represents, and how it was processed. In this section, we will explore what kind of relevant metadata is available for NGS data, and how to capture it in your <a href=06_file_structure.html>Assays or Project folders</a>.</p> <h2 id=what-is-metadata-and-why-it-is-important>What is metadata and why it is important<a class=headerlink href=#what-is-metadata-and-why-it-is-important title="Permanent link">Â¶</a></h2> <p><img alt=metadata src=images/metadata.png> <em>From <a href=https://www.ontotext.com/knowledgehub/fundamentals/metadata-fundamental/ >ontotext.com</a></em></p> <p>Imagine you have a batch of DNA sequences from different people. The raw sequences are like jigsaw puzzle pieces, and metadata is a cheat sheet that tells you which pieces fit where. It could include details like when and where the samples were collected, the lab procedures used, who did created them, and even the equipment involved, providing context and making sense of the who, what, when, where, and why.</p> <div class="admonition note"> <p class=admonition-title>Definition of metadata</p> <p>Metadata refers to data that provides information about other data. It describes various aspects of the data, such as its origin, structure, format, and context. Metadata is typically used to facilitate the organization, management, and interpretation of data, making it easier to locate, access, and understand. In essence, metadata adds valuable context and attributes to the primary data, enhancing its usability and ensuring efficient data management."</p> </div> <p>Let's think of an example that shows why metadata is extremelly important. Imagine you're in a big lab with a plethora of different datasets all saved under generic folder names. Without metadata, it would be like searching for a needle in a haystack. You'd have folders labeled 'Experiment123,' 'DataBatch42,' and so on, but zero clues about what's inside. With metadata, 'Experiment123' is not just that anymore, but 'DNA Sequencing of Human Cells, March 2023'. Providing relevant metadata converts data chaos into an organized repository, turning data exploration, interpretation, and insight extraction into a much easier journey. This is true not only for yourself or your lab, but to other researchers that might want to reuse your data. Collecting metadata from the very beginning of the research project will help tremendously to alleviate future efforts to understand the data. It can also help you make an organised collection of data, so that you do not lose any information regarding the data, or that you do not repeat an unnecessary experiment that someone else has done it before, but you could not find. That is, it helps you save money and time!</p> <div class="admonition note"> <p class=admonition-title>Benefits of collecting proper metadata</p> <ol> <li><strong>Data Context and Interpretation</strong>: Metadata provides crucial context to NGS data, offering insights into the experimental conditions, sample origins, and processing methods. This context is vital for understanding data variations, drawing accurate conclusions, and interpreting results correctly.</li> <li><strong>Data Discovery and Access</strong>: With metadata, researchers can easily locate and access specific NGS datasets within large repositories. Details like sample identifiers, experimental parameters, and timestamps help researchers quickly identify relevant data for their analyses.</li> <li><strong>Reproducibility and Collaboration</strong>: Metadata ensures that NGS experiments can be replicated and validated by others. By sharing comprehensive metadata, researchers enable colleagues to reproduce analyses, compare results, and collaborate effectively, bolstering the integrity of scientific findings.</li> <li><strong>Quality Control and Validation</strong>: Metadata supports data quality assessment by allowing researchers to track the origin and handling of NGS data. It enables the identification of potential errors or biases, helping researchers validate the accuracy and reliability of their analyses.</li> <li><strong>Long-Term Data Preservation</strong>: Properly documented metadata is essential for preserving NGS data over time. As research evolves, metadata ensures that future generations can understand and utilize archived NGS datasets, ensuring the continued impact of scientific discoveries.</li> </ol> </div> <h2 id=how-to-collect-metadata-in-your-folders>How to collect metadata in your folders<a class=headerlink href=#how-to-collect-metadata-in-your-folders title="Permanent link">Â¶</a></h2> <p>In our previous lesson, we learnt about how to organize your data into different types of folders: <code>Assays</code> and <code>Projects</code>. Both of these folders contain a metadata.yml file and a README.md file. In this section, we will check what kind of information you should collect in each of these files.</p> <h3 id=readmemd-file>README.md file<a class=headerlink href=#readmemd-file title="Permanent link">Â¶</a></h3> <p>The README.md file is a <a href=https://www.markdownguide.org/ >markdown file</a> that allows you to write a long description of the data placed in a folder. Since it is a markdown file, you are able to write in rich text format (bold, italic, include links, etc) what is inside the folder, why it was created/collected, how and when. If it is an <code>Assay</code> folder, you could include the laboratory protocol used to generate the samples, images explaining the experiment design, a summary of the results of the experiment and any sort of comments that would help to understand the context of the experiment. On the other hand, a 'Project' README file may contain a description of the project, what are its aims, why is it important, what 'Assays' is it using, how to interpret the code notebooks, a summary of the results and, again, any sort of comments that would help to understand the project.</p> <p>Here is an example of a README file for a `Project`` folder:</p> <div class=highlight><pre><span></span><code><a id=__codelineno-0-1 name=__codelineno-0-1 href=#__codelineno-0-1></a># NGS Analysis Project: Exploring Gene Expression in Human Tissues
<a id=__codelineno-0-2 name=__codelineno-0-2 href=#__codelineno-0-2></a>
<a id=__codelineno-0-3 name=__codelineno-0-3 href=#__codelineno-0-3></a>## Aims
<a id=__codelineno-0-4 name=__codelineno-0-4 href=#__codelineno-0-4></a>
<a id=__codelineno-0-5 name=__codelineno-0-5 href=#__codelineno-0-5></a>This project aims to investigate gene expression patterns across various human tissues using Next Generation Sequencing (NGS) data. By analyzing the transcriptomes of different tissues, we seek to uncover tissue-specific gene expression profiles and identify potential markers associated with specific biological functions or diseases.
<a id=__codelineno-0-6 name=__codelineno-0-6 href=#__codelineno-0-6></a>
<a id=__codelineno-0-7 name=__codelineno-0-7 href=#__codelineno-0-7></a>## Why It's Important
<a id=__codelineno-0-8 name=__codelineno-0-8 href=#__codelineno-0-8></a>
<a id=__codelineno-0-9 name=__codelineno-0-9 href=#__codelineno-0-9></a>Understanding tissue-specific gene expression is crucial for deciphering the molecular basis of health and disease. Identifying genes that are uniquely expressed in certain tissues can provide insights into tissue function, development, and potential therapeutic targets. This project contributes to our broader understanding of human biology and has implications for personalized medicine and disease research.
<a id=__codelineno-0-10 name=__codelineno-0-10 href=#__codelineno-0-10></a>
<a id=__codelineno-0-11 name=__codelineno-0-11 href=#__codelineno-0-11></a>## Datasets
<a id=__codelineno-0-12 name=__codelineno-0-12 href=#__codelineno-0-12></a>
<a id=__codelineno-0-13 name=__codelineno-0-13 href=#__codelineno-0-13></a>We have used internal datasets with IDs: RNA_humanSkin_20201030, RNA_humanBrain_20210102, RNA_humanLung_20220304.
<a id=__codelineno-0-14 name=__codelineno-0-14 href=#__codelineno-0-14></a>
<a id=__codelineno-0-15 name=__codelineno-0-15 href=#__codelineno-0-15></a>In addition, we utilized publicly available NGS datasets from the GTEx (Genotype-Tissue Expression) project, which provides comprehensive RNA-seq data across multiple human tissues. These datasets offer a wealth of information on gene expression levels and isoform variations across diverse tissues, making them ideal for our analysis.
<a id=__codelineno-0-16 name=__codelineno-0-16 href=#__codelineno-0-16></a>
<a id=__codelineno-0-17 name=__codelineno-0-17 href=#__codelineno-0-17></a>## Summary of Results
<a id=__codelineno-0-18 name=__codelineno-0-18 href=#__codelineno-0-18></a>
<a id=__codelineno-0-19 name=__codelineno-0-19 href=#__codelineno-0-19></a>Our analysis revealed distinct gene expression patterns among different human tissues. We identified tissue-specific genes enriched in brain tissues, highlighting their potential roles in neurodevelopment and function. Additionally, we found a set of genes that exhibit consistent expression across a range of tissues, suggesting their fundamental importance in basic cellular processes.
<a id=__codelineno-0-20 name=__codelineno-0-20 href=#__codelineno-0-20></a>
<a id=__codelineno-0-21 name=__codelineno-0-21 href=#__codelineno-0-21></a>Furthermore, our differential expression analysis unveiled significant changes in gene expression between healthy and diseased tissues, shedding light on potential molecular factors underlying various diseases. Overall, this project underscores the power of NGS data in unraveling intricate gene expression networks and their implications for human health.
<a id=__codelineno-0-22 name=__codelineno-0-22 href=#__codelineno-0-22></a>
<a id=__codelineno-0-23 name=__codelineno-0-23 href=#__codelineno-0-23></a>---
<a id=__codelineno-0-24 name=__codelineno-0-24 href=#__codelineno-0-24></a>
<a id=__codelineno-0-25 name=__codelineno-0-25 href=#__codelineno-0-25></a>For more details, refer to our [Jupyter Notebook](link-to-jupyter-notebook.ipynb) for the complete analysis pipeline and code.
</code></pre></div> <h3 id=metadatayml>metadata.yml<a class=headerlink href=#metadatayml title="Permanent link">Â¶</a></h3> <p>The metadata file is a <a href=https://fileinfo.com/extension/yml>yml file</a>, which is a text document that contains data formatted using a human-readable data format for data serialization. It is not mandatory that you use yml format, any other structured file format will work too, such as json files. We recommend using yml format because it is easily readable for humans, so non-coding people will have an easier time checking, writing or modifying the file if they need to.</p> <p><img alt="yaml file example" src=images/yml_example.png></p> <h2 id=metadata-fields>Metadata fields<a class=headerlink href=#metadata-fields title="Permanent link">Â¶</a></h2> <p>There is a ton of information you can collect regarding an NGS assay or a project. Some information fields are very general, such as author or date, while others are specific to the Assay or Project folder. Below, we will take a look at minimal information you should collect in each of the folders. But before that, we need to talk about controlled vocabularies/ontologies.</p> <p>Imagine this scenario: a researcher in genomics us excited to explore various NGS datasets from different human tissue samples to study gene expression patterns. However, she encounters a significant hurdle: the tissue names used in the datasets are inconsistent and lack standardized terms. Some datasets refer to "brain," while others use "cerebral cortex" or "cortical tissue." This lack of controlled vocabularies for tissue names complicates her data integration efforts, requiring her to spend additional time curating and mapping tissue labels to establish meaningful cross-dataset comparisons.</p> <p>So, what happened here? If the original creators of the NGS datasets had adopted a standardized vocabulary for tissue names, Dr. Smith could have seamlessly integrated the data without the need for extensive curation. By employing a widely accepted tissue ontology, like the Uberon Ontology, dataset contributors could have consistently used predefined terms, such as "brain" or "cerebral cortex." This practice would have not only simplified data integration but also facilitated accurate cross-dataset comparisons and enabled more reliable downstream analyses.</p> <p>In the context of NGS data, ontologies and controlled vocabularies play a pivotal role in clarifying and categorizing many concepts and information about your data. More examples are: the Gene Ontology (GO), which provides a shared vocabulary to describe gene functions, molecular processes, and cellular components, enhancing the consistency and comparability of NGS results, and Ensembl gene IDs, which are used to uniquely represent individual genes and provide a standardized way of referencing and accessing gene-related information across various species. By leveraging ontologies, researchers ensure that metadata fields capture specific details consistently across experiments, from sample sources and protocols to experimental conditions.</p> <div class="admonition note"> <p class=admonition-title>Definition of ontology</p> <p>An ontology is a formal representation of knowledge that encompasses concepts, their attributes, and the relationships between them within a particular domain or subject area. It serves as a structured framework for organizing and categorizing information, facilitating the interpretation, sharing, and integration of knowledge across diverse applications and disciplines. Ontologies employ standardized vocabularies and define the semantics of terms, enabling effective communication and reasoning among humans and computer systems. They play a pivotal role in knowledge representation, data integration, and semantic interoperability, contributing to enhanced understanding, collaboration, and analysis within complex domains.</p> </div> <p>This standardization not only enhances data discoverability and interoperability but also empowers robust data analysis, accelerates knowledge sharing, and enables meaningful cross-study comparisons. In essence, ontologies serve as the universal translators of the scientific language, fostering a harmonious symphony of data interpretation and collaboration.</p> <h3 id=general-metadata-fields>General metadata fields<a class=headerlink href=#general-metadata-fields title="Permanent link">Â¶</a></h3> <p>Here you can find a list of suggestions for general metadata fields that can be used for both assays and project folders:</p> <ul> <li><strong>Title</strong>: A brief yet informative name for the dataset.</li> <li><strong>Author(s)</strong>: The individual(s) or organization responsible for creating the dataset. You can use your <a href=https://orcid.org/ >ORCID</a></li> <li><strong>Date Created</strong>: The date when the dataset was originally generated or compiled. Use YYYY-MM-DD format!</li> <li><strong>Date Modified</strong>: The date when the dataset was last updated or modified. Use YYYY-MM-DD format!</li> <li><strong>Object ID</strong>: The project or assay ID for tracking and reference purposes.</li> <li><strong>Description</strong>: A short narrative explaining the content, purpose, and context.</li> <li><strong>Keywords</strong>: A set of descriptive terms or phrases that capture the folder's main topics and attributes.</li> <li><strong>Ethical and Legal Considerations</strong>: Information about ethical approvals, consent, and any legal restrictions.</li> <li><strong>Version</strong>: The version number or identifier for the folder, useful for tracking changes.</li> <li><strong>Related Publications</strong>: Links or references to any scientific publications associated with the folder. Try to add here the DOI!</li> <li><strong>Funding Source</strong>: Details about the funding agency or source that supported the research and data generation.</li> <li><strong>License</strong>: The type of license or terms of use associated with the dataset/project.</li> <li><strong>Contact Information</strong>: Contact details for individuals who can provide further information about the dataset/project.</li> </ul> <h3 id=assay-metadata-fields>Assay metadata fields<a class=headerlink href=#assay-metadata-fields title="Permanent link">Â¶</a></h3> <p>Here you will find a table with possible metadata fields that you can use to annotate and track your <code>Assay</code> folders:</p> <table> <thead> <tr> <th style="text-align: left;">Metadata field</th> <th style="text-align: left;">Definition</th> <th style="text-align: left;">Format</th> <th style="text-align: left;">Ontology</th> <th style="text-align: left;">Example</th> </tr> </thead> <tbody> <tr> <td style="text-align: left;">assay_ID</td> <td style="text-align: left;">Identifier for the assay that is at least unique within the project</td> <td style="text-align: left;">&lt;Assay-ID&gt;_&lt;keyword&gt;_YYYYMMDD</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">CHIP_Oct4_20200101</td> </tr> <tr> <td style="text-align: left;">assay_type</td> <td style="text-align: left;">The type of experiment performed, eg ATAC-seq or seqFISH</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">ontology field- e.g. EFO or OBI</td> <td style="text-align: left;">ChIPseq</td> </tr> <tr> <td style="text-align: left;">assay_subtype</td> <td style="text-align: left;">More specific type or assay like bulk nascent RNAseq or single cell ATACseq</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">ontology field- e.g. EFO or OBI</td> <td style="text-align: left;">bulk ChIPseq</td> </tr> <tr> <td style="text-align: left;">owner</td> <td style="text-align: left;">Owner of the assay (who made the experiment?).</td> <td style="text-align: left;">&lt;First Name&gt; &lt;Last Name&gt;</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">Jose Romero</td> </tr> <tr> <td style="text-align: left;">platform</td> <td style="text-align: left;">The type of instrument used to perform the assay, eg Illumina HiSeq 4000 or Fluidigm C1 microfluidics platform</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">ontology field- e.g. EFO or OBI</td> <td style="text-align: left;">Illumina</td> </tr> <tr> <td style="text-align: left;">extraction_method</td> <td style="text-align: left;">Technique used to extract the nucleic acid from the cell</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">ontology field- e.g. EFO or OBI</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">library_method</td> <td style="text-align: left;">Technique used to amplify a cDNA library</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">ontology field- e.g. EFO or OBI</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">external_accessions</td> <td style="text-align: left;">Accession numbers from external resources to which assay or protocol information was submitted</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">eg protocols.io, AE, GEO accession number, etc</td> <td style="text-align: left;">GSEXXXXX</td> </tr> <tr> <td style="text-align: left;">keyword</td> <td style="text-align: left;">Keyword for easy identification</td> <td style="text-align: left;">wordWord</td> <td style="text-align: left;">camelCase</td> <td style="text-align: left;">Oct4ChIP</td> </tr> <tr> <td style="text-align: left;">date</td> <td style="text-align: left;">Date of assay creation</td> <td style="text-align: left;">YYYYMMDD</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">20200101</td> </tr> <tr> <td style="text-align: left;">nsamples</td> <td style="text-align: left;">Number of samples analyzed in this assay</td> <td style="text-align: left;">&lt;integer&gt;</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">9</td> </tr> <tr> <td style="text-align: left;">is_paired</td> <td style="text-align: left;">Paired fastq files or not</td> <td style="text-align: left;">&lt;single OR paired&gt;</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">single</td> </tr> <tr> <td style="text-align: left;">pipeline</td> <td style="text-align: left;">Pipeline used to process data and version</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nf-core/chipseq -r 1.0</td> </tr> <tr> <td style="text-align: left;">strandedness</td> <td style="text-align: left;">The strandedness of the cDNA library</td> <td style="text-align: left;">&lt;+ OR - OR *&gt;</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">*</td> </tr> <tr> <td style="text-align: left;">processed_by</td> <td style="text-align: left;">Who processed the data</td> <td style="text-align: left;">&lt;First Name&gt; &lt;Last Name&gt;</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">Sarah Lundregan</td> </tr> <tr> <td style="text-align: left;">organism</td> <td style="text-align: left;">Organism origin</td> <td style="text-align: left;">&lt;Genus species&gt;</td> <td style="text-align: left;">Taxonomy name</td> <td style="text-align: left;">Mus musculus</td> </tr> <tr> <td style="text-align: left;">origin</td> <td style="text-align: left;">Is internal or external (from a public resources) data</td> <td style="text-align: left;">&lt;internal OR external&gt;</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">internal</td> </tr> <tr> <td style="text-align: left;">path</td> <td style="text-align: left;">Path to files</td> <td style="text-align: left;">&lt;/path/to/file&gt;</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">short_desc</td> <td style="text-align: left;">Short description of the assay</td> <td style="text-align: left;">plain text</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">Oct4 ChIP after pERK activation</td> </tr> <tr> <td style="text-align: left;">ELN_ID</td> <td style="text-align: left;">ID of the experiment/assay in your Electronic Lab Notebook software, like labguru or benchling</td> <td style="text-align: left;">plain text</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> </tr> </tbody> </table> <h3 id=sample-metadata-fields>Sample metadata fields<a class=headerlink href=#sample-metadata-fields title="Permanent link">Â¶</a></h3> <p>There is some information that will be specific to your samples. For example, which samples are treated, which are control, which tissue do they come from, which cell type, etc. In this case, it would be beneficial if you include all this information in the <a href=06_file_structure.html><code>samplesheet.csv</code></a> that describes the fastq files. Here is a list of possible metadata fields that you can use:</p> <table> <thead> <tr> <th style="text-align: left;">Metadata field</th> <th style="text-align: left;">Definition</th> <th style="text-align: left;">Format</th> <th style="text-align: left;">Ontology</th> <th style="text-align: left;">Example</th> </tr> </thead> <tbody> <tr> <td style="text-align: left;">sample</td> <td style="text-align: left;">Name of the sample</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">control_rep1, treat_rep1</td> </tr> <tr> <td style="text-align: left;">fastq_1</td> <td style="text-align: left;">Path to fastq file 1</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">AEG588A1_S1_L002_R1_001.fastq.gz</td> </tr> <tr> <td style="text-align: left;">fastq_2</td> <td style="text-align: left;">Path to paired fastq file, if it is a paired experiment</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">AEG588A1_S1_L002_R2_001.fastq.gz</td> </tr> <tr> <td style="text-align: left;">strandedness</td> <td style="text-align: left;">The strandedness of the cDNA library</td> <td style="text-align: left;">&lt;unstranded OR forward OR reverse &gt;</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">unstranded</td> </tr> <tr> <td style="text-align: left;">condition</td> <td style="text-align: left;">Variable of interest of the experiment, such as "control", "treatment", etc</td> <td style="text-align: left;">wordWord</td> <td style="text-align: left;">camelCase</td> <td style="text-align: left;">control, treat1, treat2</td> </tr> <tr> <td style="text-align: left;">cell_type</td> <td style="text-align: left;">The cell type(s) known or selected to be present in the sample</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">ontology field- e.g. EFO or OBI</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">tissue</td> <td style="text-align: left;">The tissue from which the sample was taken</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">Uberon</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">sex</td> <td style="text-align: left;">The biological/genetic sex of the sample</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">ontology field- e.g. EFO or OBI</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">cell_line</td> <td style="text-align: left;">Cell line of the sample</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">ontology field- e.g. EFO or OBI</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">organism</td> <td style="text-align: left;">Organism origin of the sample</td> <td style="text-align: left;"><genus species></genus></td> <td style="text-align: left;">Taxonomy</td> <td style="text-align: left;">Mus musculus</td> </tr> <tr> <td style="text-align: left;">replicate</td> <td style="text-align: left;">Replicate number</td> <td style="text-align: left;">&lt;integer&gt;</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">1</td> </tr> <tr> <td style="text-align: left;">batch</td> <td style="text-align: left;">Batch information</td> <td style="text-align: left;">wordWord</td> <td style="text-align: left;">camelCase</td> <td style="text-align: left;">1</td> </tr> <tr> <td style="text-align: left;">disease</td> <td style="text-align: left;">Any diseases that may affect the sample</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">Disease Ontology or MONDO</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">developmental_stage</td> <td style="text-align: left;">The developmental stage of the sample</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">sample_type</td> <td style="text-align: left;">The type of the collected specimen, eg tissue biopsy, blood draw or throat swab</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">strain</td> <td style="text-align: left;">Strain of the species from which the sample was collected, if applicable</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">ontology field - e.g. NCBITaxonomy</td> <td style="text-align: left;">nan</td> </tr> <tr> <td style="text-align: left;">genetic variation</td> <td style="text-align: left;">Any relevant genetic differences from the specimen or sample to the expected genomic information for this species, eg abnormal chromosome counts, major translocations or indels</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> <td style="text-align: left;">nan</td> </tr> </tbody> </table> <h3 id=project-metadata-fields>Project metadata fields<a class=headerlink href=#project-metadata-fields title="Permanent link">Â¶</a></h3> <p>Here you will find a table with possible metadata fields that you can use to annotate and track your <code>Project</code> folders:</p> <p>In development.</p> <table> <thead> <tr> <th style="text-align: left;">Metadata field</th> <th style="text-align: left;">Definition</th> <th style="text-align: left;">Format</th> <th style="text-align: right;">Ontology</th> <th style="text-align: left;">Example</th> </tr> </thead> <tbody> <tr> <td style="text-align: left;">project</td> <td style="text-align: left;">Project ID</td> <td style="text-align: left;">&lt;surname&gt;_et_al_2023</td> <td style="text-align: right;">nan</td> <td style="text-align: left;">proks_et_al_2023</td> </tr> <tr> <td style="text-align: left;">author</td> <td style="text-align: left;">Owner of the project</td> <td style="text-align: left;">&lt;First name&gt; &lt;Surname&gt;</td> <td style="text-align: right;">nan</td> <td style="text-align: left;">Martin Proks</td> </tr> <tr> <td style="text-align: left;">date</td> <td style="text-align: left;">Date of creation</td> <td style="text-align: left;">YYYYMMDD</td> <td style="text-align: right;">nan</td> <td style="text-align: left;">20230101</td> </tr> <tr> <td style="text-align: left;">description</td> <td style="text-align: left;">Short description of the project</td> <td style="text-align: left;">Plain text</td> <td style="text-align: right;">nan</td> <td style="text-align: left;">This is a project describing the effect of Oct4 perturbation after pERK activation</td> </tr> </tbody> </table> <h2 id=more-info>More info<a class=headerlink href=#more-info title="Permanent link">Â¶</a></h2> <p>The information provided in this lesson is not at all exhaustive. There might be many more fields and controlled vocabularies that could be useful for your NGS data. We recommend that you take a look at the following sources for more information!</p> <ul> <li><a href=https://faircookbook.elixir-europe.org/content/recipes/interoperability/transcriptomics-metadata.html#analysis-metadata>Transcriptomics metadata standards and fields</a></li> <li><a href=https://lamin.ai/docs/bionty>Bionty</a>: Biological ontologies for data scientists.</li> </ul> <div class="admonition question"> <p class=admonition-title>Exercise: modify the metadata files in your cookiecutter templates</p> <p>We have seen some examples of metadata for NGS data. It is time now to customize your cookiecutter templates and modify the metadata.yml files so that they fit your needs! </p> <ol> <li>Think about what kind of metadata you would like to include.</li> <li>Modify the <code>cookiecutter.json</code> file so that when you create a new folder template, all the metadata is filled accordingly. <img alt=cookiecutter_json_example src=images/cookiecutter_json.png></li> <li>Modify the <code>metadata.yml</code> file so that it includes the metadata recorded by the <code>cookiecutter.json</code> file. <img alt=assay_metadata_example src=images/assay_metadata.png></li> <li>Modify the <code>README.md</code> file so that it includes the short description recorded by the <code>cookiecutter.json</code> file.</li> <li>Git add, commit and push the changes of your template.</li> <li>Test your folders by using the command <code>cookiecutter &lt;URL to your cookiecutter repository in GitHub&gt;</code></li> </ol> </div> <h2 id=wrap-up>Wrap up<a class=headerlink href=#wrap-up title="Permanent link">Â¶</a></h2> <p>In this lesson we have learned about what metadata should be attached to your data in order to be reusable and understood in the future. Not only that, the metadata provided can help to process your samples adequately, and could even be useful for metadata studies. We have also briefly introduced some controlled vocabularies and sources for different fields, such as Disease Ontologies, Cell type ontologies, Organisms, etc. However, since this course is not aimed at enforcing the use of these vocabularies (that would be very complex), their implementation and use is up to you! In the next lesson we will learn how we can collect all the metadata information in each of the folders in order to make a database of assays and projects, allowing you to browse all your data so that they are always at hand!</p></div> <hr> <div class=md-source-file> <small> Last update: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">November 30, 2023</span> <br> Created: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">November 30, 2023</span> </small> </div> </article> </div> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </button> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=06_file_structure.html class="md-footer__link md-footer__link--prev" aria-label="Previous: File structure and naming conventions"> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <span class=md-footer__direction> Previous </span> <div class=md-ellipsis> File structure and naming conventions </div> </div> </a> <a href=08_database.html class="md-footer__link md-footer__link--next" aria-label="Next: Database of your data and projects"> <div class=md-footer__title> <span class=md-footer__direction> Next </span> <div class=md-ellipsis> Database of your data and projects </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> CC-BY-SA 4.0 license </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=https://github.com/hds-sandbox target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://twitter.com/ucph_heads target=_blank rel=noopener title=twitter.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": ".", "features": ["content.code.annotate", "content.tooltips", "navigation.indexes", "navigation.sections", "navigation.footer", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow"], "search": "assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script> <script src=assets/javascripts/bundle.cd18aaf1.min.js></script> <script src=javascripts/mathjax.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>