<!doctype html><html lang=en class=no-js> <head><script type=application/ld+json>
    {
      "@context": "https://schema.org",
      "@id": "https://hds-sandbox.github.io/RDM_NGS_course/",
      "@type": "LearningResource",
      "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
      "description": "This workshop a basic intro to RDM, and how to apply good practices to your NGS data",
      "keywords": [
        "RDM, Research Data Management, NGS, Bioinformatics, Data Management Plan"
      ],
      "learningResourceType": [
        "e-learning"
      ],
      "license": [
        {
          "@id": "# Link to license webpage (https://creativecommons.org/licenses/by-sa/4.0/)",
          "@type": "CreativeWork"
        }
      ],
      "mentions": [
        {
          "@id": "# Link to DOI or zenodo materials (10.5281/zenodo.7660225)",
          "@type": "Thing"
        },
        {
          "@id": "#Another link to DOI or zenodo materials (10.5281/zenodo.7565997)",
          "@type": "Thing"
        }
      ],
      "name": "Introduction to NGS Data Management",
      "educationalLevel": "Beginner",
      "audience": "PhD, postdocs, MSc, etc"
    }
    </script><title>Repositories for NGS - RDM for NGS course</title><link rel=stylesheet href=assets/stylesheets/main.fad675c6.min.css><link rel=stylesheet href=assets/stylesheets/palette.356b1318.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=stylesheets/neoteroi-mkdocs.css><link rel=stylesheet href=stylesheets/extra.css><script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script id=__analytics>function __md_analytics(){function n(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],n("js",new Date),n("config",""),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&n("event","search",{search_term:this.value})}),document$.subscribe(function(){var a=document.forms.feedback;if(void 0!==a)for(var e of a.querySelectorAll("[type=submit]"))e.addEventListener("click",function(e){e.preventDefault();var t=document.location.pathname,e=this.getAttribute("data-md-value");n("event","feedback",{page:t,data:e}),a.firstElementChild.disabled=!0;e=a.querySelector(".md-feedback__note [data-md-value='"+e+"']");e&&(e.hidden=!1)}),a.hidden=!1}),location$.subscribe(function(e){n("config","",{page_path:e.pathname})})});var e=document.createElement("script");e.async=!0,e.src="https://www.googletagmanager.com/gtag/js?id=",document.getElementById("__analytics").insertAdjacentElement("afterEnd",e)}</script><script>"undefined"!=typeof __md_analytics&&__md_analytics()</script></head> <body dir=ltr data-md-color-scheme=brightness data-md-color-primary=indigo data-md-color-accent=indigo> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#repositories-for-ngs-data class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class="md-header md-header--shadow md-header--lifted" data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=http://hds-sandbox.github.io/ title="RDM for NGS course" class="md-header__button md-logo" aria-label="RDM for NGS course" data-md-component=logo> <img src=images/logo.png alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> RDM for NGS course </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Repositories for NGS </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=brightness data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to dark mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_2 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=indigo data-md-color-accent=indigo aria-label="Switch to light mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" title=Share aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"/></svg> </a> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/hds-sandbox/RDM_NGS_course title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> RDM_NGS_course </div> </a> </div> </nav> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=index.html class=md-tabs__link> Introduction </a> </li> <li class="md-tabs__item md-tabs__item--active"> <a href=01_RDM_intro.html class=md-tabs__link> Course contents </a> </li> <li class=md-tabs__item> <a href=practical_workshop.html class=md-tabs__link> Practical workshop </a> </li> <li class=md-tabs__item> <a href=keywords.html class=md-tabs__link> Keyword index </a> </li> <li class=md-tabs__item> <a href=contributors.html class=md-tabs__link> Contributors </a> </li> </ul> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=http://hds-sandbox.github.io/ title="RDM for NGS course" class="md-nav__button md-logo" aria-label="RDM for NGS course" data-md-component=logo> <img src=images/logo.png alt=logo> </a> RDM for NGS course </label> <div class=md-nav__source> <a href=https://github.com/hds-sandbox/RDM_NGS_course title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> RDM_NGS_course </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=index.html class=md-nav__link> <span class=md-ellipsis> Introduction </span> </a> </li> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_2 checked> <label class=md-nav__link for=__nav_2 id=__nav_2_label tabindex> <span class=md-ellipsis> Course contents </span> <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_2_label aria-expanded=true> <label class=md-nav__title for=__nav_2> <span class="md-nav__icon md-icon"></span> Course contents </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=01_RDM_intro.html class=md-nav__link> <span class=md-ellipsis> What is RDM </span> </a> </li> <li class=md-nav__item> <a href=02_NGS_data.html class=md-nav__link> <span class=md-ellipsis> NGS data </span> </a> </li> <li class=md-nav__item> <a href=03_data_life_cycle.html class=md-nav__link> <span class=md-ellipsis> Data Life Cycle </span> </a> </li> <li class=md-nav__item> <a href=04_OS_FAIR.html class=md-nav__link> <span class=md-ellipsis> Open Science and FAIR principles </span> </a> </li> <li class=md-nav__item> <a href=05_DMP.html class=md-nav__link> <span class=md-ellipsis> Data Management Plans </span> </a> </li> <li class=md-nav__item> <a href=06_file_structure.html class=md-nav__link> <span class=md-ellipsis> File structure and naming conventions </span> </a> </li> <li class=md-nav__item> <a href=07_metadata.html class=md-nav__link> <span class=md-ellipsis> Metadata applied to NGS data </span> </a> </li> <li class=md-nav__item> <a href=08_database.html class=md-nav__link> <span class=md-ellipsis> Database of your data and projects </span> </a> </li> <li class=md-nav__item> <a href=09_version_control.html class=md-nav__link> <span class=md-ellipsis> Version control of your data analysis </span> </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> <span class=md-ellipsis> Archiving your data </span> <span class="md-nav__icon md-icon"></span> </label> <a href=10_repos.html class="md-nav__link md-nav__link--active"> <span class=md-ellipsis> Archiving your data </span> </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#what-is-a-repositoryarchive class=md-nav__link> <span class=md-ellipsis> What is a repository/archive? </span> </a> </li> <li class=md-nav__item> <a href=#why-are-they-important class=md-nav__link> <span class=md-ellipsis> Why are they important? </span> </a> </li> <li class=md-nav__item> <a href=#types-of-repositories class=md-nav__link> <span class=md-ellipsis> Types of repositories </span> </a> <nav class=md-nav aria-label="Types of repositories"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#domain-specific-repositories-for-ngs-data-geo-or-annotare class=md-nav__link> <span class=md-ellipsis> Domain specific repositories for NGS data: GEO or Annotare </span> </a> <nav class=md-nav aria-label="Domain specific repositories for NGS data: GEO or Annotare"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#gene-expression-omnibus class=md-nav__link> <span class=md-ellipsis> Gene Expression Omnibus </span> </a> </li> <li class=md-nav__item> <a href=#annotare class=md-nav__link> <span class=md-ellipsis> Annotare </span> </a> </li> <li class=md-nav__item> <a href=#ngs-data-upload-to-geo-or-annotare class=md-nav__link> <span class=md-ellipsis> NGS Data upload to GEO or Annotare </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#a-general-repository-zenodo class=md-nav__link> <span class=md-ellipsis> A general repository: Zenodo </span> </a> <nav class=md-nav aria-label="A general repository: Zenodo"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#project-archiving-in-zenodo class=md-nav__link> <span class=md-ellipsis> Project archiving in Zenodo </span> </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#wrap-up class=md-nav__link> <span class=md-ellipsis> Wrap up </span> </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=practical_workshop.html class=md-nav__link> <span class=md-ellipsis> Practical workshop </span> </a> </li> <li class=md-nav__item> <a href=keywords.html class=md-nav__link> <span class=md-ellipsis> Keyword index </span> </a> </li> <li class=md-nav__item> <a href=contributors.html class=md-nav__link> <span class=md-ellipsis> Contributors </span> </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#what-is-a-repositoryarchive class=md-nav__link> <span class=md-ellipsis> What is a repository/archive? </span> </a> </li> <li class=md-nav__item> <a href=#why-are-they-important class=md-nav__link> <span class=md-ellipsis> Why are they important? </span> </a> </li> <li class=md-nav__item> <a href=#types-of-repositories class=md-nav__link> <span class=md-ellipsis> Types of repositories </span> </a> <nav class=md-nav aria-label="Types of repositories"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#domain-specific-repositories-for-ngs-data-geo-or-annotare class=md-nav__link> <span class=md-ellipsis> Domain specific repositories for NGS data: GEO or Annotare </span> </a> <nav class=md-nav aria-label="Domain specific repositories for NGS data: GEO or Annotare"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#gene-expression-omnibus class=md-nav__link> <span class=md-ellipsis> Gene Expression Omnibus </span> </a> </li> <li class=md-nav__item> <a href=#annotare class=md-nav__link> <span class=md-ellipsis> Annotare </span> </a> </li> <li class=md-nav__item> <a href=#ngs-data-upload-to-geo-or-annotare class=md-nav__link> <span class=md-ellipsis> NGS Data upload to GEO or Annotare </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#a-general-repository-zenodo class=md-nav__link> <span class=md-ellipsis> A general repository: Zenodo </span> </a> <nav class=md-nav aria-label="A general repository: Zenodo"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#project-archiving-in-zenodo class=md-nav__link> <span class=md-ellipsis> Project archiving in Zenodo </span> </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#wrap-up class=md-nav__link> <span class=md-ellipsis> Wrap up </span> </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <div><h1 id=repositories-for-ngs-data>Repositories for NGS data<a class=headerlink href=#repositories-for-ngs-data title="Permanent link">Â¶</a></h1> <p><strong>Last updated:</strong> <em><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">November 30, 2023</span></em></p> <div class="admonition note"> <p class=admonition-title>Section Overview</p> <p>ðŸ•° <strong>Time Estimation:</strong> X minutes </p> <p>ðŸ’¬ <strong>Learning Objectives:</strong> </p> <ol> <li>Learn what are repositories and which ones are relevant for NGS data</li> <li>Learn where to archive your GitHub data analysis repositories</li> </ol> </div> <p>In this lesson, we're going to explore repositories like Zenodo, Gene Expression Omnibus, and Annotare. While platforms like GitHub are great for version control and collaborative coding, these repositories serve a different purpose. They're designed specifically for archiving and sharing scientific data, ensuring it's preserved for the long term and accessible to the global research community. You could think of them as secure digital libraries for your valuable NGS data.</p> <h2 id=what-is-a-repositoryarchive>What is a repository/archive?<a class=headerlink href=#what-is-a-repositoryarchive title="Permanent link">Â¶</a></h2> <p>Specialized repositories/archives are dedicated digital platforms designed for the secure storage, curation, and dissemination of scientific data. These repositories hold great importance in the research community as they serve as reliable archives for preserving valuable datasets. Their standardized formats and robust curation processes ensure the long-term accessibility and citability of research findings. Researchers worldwide rely on these repositories to share, discover, and validate scientific information, thereby fostering transparency, collaboration, and the advancement of knowledge across various domains of study.</p> <h2 id=why-are-they-important>Why are they important?<a class=headerlink href=#why-are-they-important title="Permanent link">Â¶</a></h2> <p>Archiving your data in these repositories offers a multitude of benefits. Firstly, it ensures the enduring accessibility and preservation of your research, safeguarding it for future generations of scientists. Additionally, repositories typically provide a unique Identifier (often a DOI) for your dataset, granting it a citable status in the academic world. This not only enhances the visibility and impact of your work but also facilitates proper attribution.</p> <p>Moreover, these repositories often encourage or require comprehensive metadata to accompany your data. This rich contextual information includes details about the methodology, experimental setup, and any other pertinent information. Such metadata greatly enhances the discoverability and interpretability of your dataset, enabling fellow researchers to effectively use and build upon your work. In essence, archiving your data in these repositories not only fulfills scholarly obligations but also amplifies the reach and influence of your research in the scientific community.</p> <p>In addition, depositing data in these archives is often a mandatory requirement set by scientific journals and funding agencies. This reflects the growing recognition of the critical role these repositories play in ensuring transparency, reproducibility, and the integrity of research outcomes. By adhering to these guidelines, researchers contribute to the broader scientific community and increase the quality of their research.</p> <h2 id=types-of-repositories>Types of repositories<a class=headerlink href=#types-of-repositories title="Permanent link">Â¶</a></h2> <p>We could divide archives in two mayor categories, general or domain-specific. General repositories like Zenodo cater to a wide range of disciplines, providing a versatile platform for data archiving. On the other hand, domain-specific repositories are tailored to particular scientific fields, like GEO and Annotare in the field of NGS, offering specialized curation and context-specific features. Utilizing domain-specific repositories often provides researchers with a more targeted audience, deeper domain expertise, and enhanced visibility within their specific research community. This focused approach ensures that data is curated and contextualized in a way that aligns closely with the standards and expectations of that particular field, maximizing its impact and utility.</p> <h3 id=domain-specific-repositories-for-ngs-data-geo-or-annotare>Domain specific repositories for NGS data: GEO or Annotare<a class=headerlink href=#domain-specific-repositories-for-ngs-data-geo-or-annotare title="Permanent link">Â¶</a></h3> <h4 id=gene-expression-omnibus>Gene Expression Omnibus<a class=headerlink href=#gene-expression-omnibus title="Permanent link">Â¶</a></h4> <p>The Gene Expression Omnibus, commonly known as GEO, is a specialized repository curated by the National Center for Biotechnology Information (NCBI). It is dedicated to archiving and sharing high-throughput functional genomic data sets, primarily focused on gene expression data.</p> <p>Researchers can easily deposit and access a wide range of genomic data, including microarray and high-throughput sequencing studies. GEO provides a structured platform for researchers to share their findings with the scientific community, enhancing data transparency and reproducibility.</p> <p>GEO assigns unique accession numbers to each dataset, ensuring traceability and enabling proper citation in research publications. Its domain-specific focus on functional genomics makes it an invaluable resource for researchers in genetics, genomics, and related fields, allowing for the comprehensive exploration of gene expression patterns across various biological conditions and experimental designs.</p> <h4 id=annotare>Annotare<a class=headerlink href=#annotare title="Permanent link">Â¶</a></h4> <p>Annotare is a specialized repository hosted by the European Bioinformatics Institute (EBI) that is tailored for the submission and storage of functional genomics experiments, particularly those involving high-throughput sequencing data. Unlike general repositories, Annotare provides a domain-specific platform optimized for researchers in the field of functional genomics.</p> <p>Researchers can upload their experimental data along with comprehensive metadata, ensuring that the context and details of the experiment are preserved. Annotare supports a wide range of genomic data types, making it a versatile platform for archiving diverse functional genomics studies.</p> <p>By focusing specifically on functional genomics, Annotare offers researchers a curated environment that aligns closely with the standards and practices of this specialized field. This ensures that data is stored and curated in a manner that is most useful for researchers in the genomics community. The platform's specialization enhances data discoverability, promotes collaboration, and facilitates deeper insights into the functional aspects of the genome.</p> <h4 id=ngs-data-upload-to-geo-or-annotare>NGS Data upload to GEO or Annotare<a class=headerlink href=#ngs-data-upload-to-geo-or-annotare title="Permanent link">Â¶</a></h4> <p><a href=https://www.ncbi.nlm.nih.gov/geo/ >GEO</a> and <a href=https://www.ebi.ac.uk/fg/annotare/login/ >Annotare</a> are excellent repository choices to deposit your NGS data. Both Annotare and GEO adhere to established community standards for data submission and sharing in the field of functional genomics:</p> <ol> <li><a href=https://pubmed.ncbi.nlm.nih.gov/11726920/ ><strong>Minimum Information About a Microarray Experiment (MIAME)</strong></a>: This is a set of guidelines established to ensure the comprehensive and standardized reporting of microarray experiments. Both Annotare and GEO require compliance with MIAME standards for microarray data submissions.</li> <li><a href=https://www.fged.org/projects/minseqe/ ><strong>Minimum Information about a high-throughput SeQuencing Experiment (MIxS)</strong></a>: MIxS is a set of standards developed by the Genomic Standards Consortium to ensure consistent reporting of metadata for high-throughput sequencing experiments. Annotare and GEO require adherence to MIxS standards for sequencing data submissions.</li> <li><a href=https://www.ncbi.nlm.nih.gov/sra/docs/submit/ ><strong>Sequence Read Archive (SRA) Submission Guidelines</strong></a>: Both Annotare and GEO follow the submission guidelines set forth by the Sequence Read Archive, which include requirements for data formatting, metadata inclusion, and quality control.</li> <li><strong>Community-Specific Standards</strong>: In addition to the above, Annotare and GEO may also adhere to community-specific standards and guidelines established by the functional genomics research community. These standards are designed to ensure that submitted data meets the specific requirements and expectations of the field.</li> </ol> <p>By adhering to these standards, Annotare and GEO ensure that the data submitted to their repositories is of high quality, well-documented, and compliant with community best practices. This facilitates data discovery, reproducibility, and interoperability within the scientific community.</p> <p>These repositories will only accept NGS data and information related to the creation of the data. This includes the raw FASTQ files, sample metadata, including protocols and descriptions of how the samples and data where processed, as well as final pre-processing results such as read count matrices or genomic position files (like BED). If you adhere to the <code>Assay</code> folder creation guideline of <a href=06_file_structure.html>lesson 6</a>, you will have a very easy time filling up the required documentation and information needed to submit the data in your <code>Assay</code> folder to one of these repositories.</p> <p>Nonetheless, the repositories will not accept other data created by your down-stream analyses, neither the code used for data analyses! This means anything that you have done in your <code>Project</code> folder. However, your <code>Project</code> folder is already version controlled by GitHub (<a href=09_version_control.html>see previous lesson</a>), so there is no need to worry. We will see in the section below how to archive your `Project`` folder as well using a general repository like Zenodo.</p> <h3 id=a-general-repository-zenodo>A general repository: Zenodo<a class=headerlink href=#a-general-repository-zenodo title="Permanent link">Â¶</a></h3> <p>Zenodo[https://zenodo.org/] is an open-access digital repository designed to facilitate the archiving of scientific research outputs. It operates under the umbrella of the European Organization for Nuclear Research (CERN) and is supported by the European Commission. Zenodo accommodates a broad spectrum of research outputs, including datasets, papers, software, and multimedia files. This versatility makes it an invaluable resource for researchers across a wide array of domains, promoting transparency, collaboration, and the advancement of knowledge on a global scale.</p> <p>Operating on a user-friendly web platform, Zenodo allows researchers to easily upload, share, and preserve their research data and related materials. Upon deposit, each item is assigned a unique Digital Object Identifier (DOI), granting it a citable status and ensuring its long-term accessibility. Additionally, Zenodo provides robust metadata capabilities, enabling researchers to enrich their submissions with detailed contextual information. In addition, it allows you to <a href=https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content>link you GitHub account</a>, providing a streamlined way to archive a specific release of your GitHub repository directly into Zenodo. This integration simplifies the process of preserving a snapshot of your project's progress for long-term accessibility and citation.</p> <h4 id=project-archiving-in-zenodo><code>Project</code> archiving in Zenodo<a class=headerlink href=#project-archiving-in-zenodo title="Permanent link">Â¶</a></h4> <p>Once your <a href=https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content>accounts are linked</a>, creating a Zenodo archive is as simple as tagging a release in your GitHub repository. Zenodo will automatically detect the release and generate a corresponding archive. This archive is assigned a unique Digital Object Identifier (DOI), making it a citable reference for your work. So, before submitting your work in a journal, make sure to link your data analysis repository to <a href=https://zenodo.org/ >Zenodo</a>, get a DOI and cite it in your manuscript!</p> <p>By leveraging this integration, you ensure that significant milestones in your project are preserved in a reliable and accessible manner. This not only facilitates proper attribution but also contributes to the broader scientific community's ability to reproduce and build upon your research.</p> <div class="admonition question"> <p class=admonition-title>Exercise 6: Archive a <code>Project</code> GitHub repo in Zenodo</p> <ol> <li>In order to archive your GitHub repos in Zenodo, you will first need to <a href=https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content>link your Zenodo and GitHub accounts</a>.</li> <li>Once your accounts are linked, go to your Zenodo GitHub account settings and turn on the GitHub repository you want to archive. <img alt=zenodo_github_link src=images/zenodo_github.png></li> <li>Creating a Zenodo archive is now as simple as <a href=https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository>making a release</a> in your GitHub repository. Remember to make a proper tag! <strong>NOTE: If you make a release before enabling the GitHub repository in Zenodo, it will not appear in Zenodo!</strong> <img alt=github_release src=images/github_release.png></li> <li>Zenodo will automatically detect the release and it should appear in your Zenodo upload page. <img alt=zenodo_archives src=images/zenodo_archives.png></li> <li>This archive is assigned a unique Digital Object Identifier (DOI), making it a citable reference for your work. <img alt=zenodo_example src=images/zenodo_example.png></li> </ol> <p>Before submitting your work in a journal, make sure to link your data analysis repository to <a href=https://zenodo.org/ >Zenodo</a>, get a DOI and cite it in your manuscript!</p> </div> <h2 id=wrap-up>Wrap up<a class=headerlink href=#wrap-up title="Permanent link">Â¶</a></h2> <p>In this final lesson we have learned how to wrap up a project/manuscript experiment by submitting your data to a domain-specific repository, while archiving your data analysis GitHub repositories in Zenodo. By following the simple lessons shown in this workshop, you will dramatically improve the FAIRability of your data, as well as organizing and structuring it in a way that will be much more useful in the future. This advantages do not serve yourself only, but your teammates, group leader and the general scientific population!</p> <p>We hope that you found this workshop useful. If you would like to leave us some comments or suggestions, feel free to answer this form! </p> <p><a href=https://forms.office.com/e/BZkpzDKL0L><strong>FEEDBACK FORM</strong></a></p></div> <hr> <div class=md-source-file> <small> Last update: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">November 30, 2023</span> <br> Created: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-date">November 30, 2023</span> </small> </div> </article> </div> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </button> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=09_version_control.html class="md-footer__link md-footer__link--prev" aria-label="Previous: Version control of your data analysis"> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <span class=md-footer__direction> Previous </span> <div class=md-ellipsis> Version control of your data analysis </div> </div> </a> <a href=practical_workshop.html class="md-footer__link md-footer__link--next" aria-label="Next: Practical workshop"> <div class=md-footer__title> <span class=md-footer__direction> Next </span> <div class=md-ellipsis> Practical workshop </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> CC-BY-SA 4.0 license </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=https://github.com/hds-sandbox target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://twitter.com/ucph_heads target=_blank rel=noopener title=twitter.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": ".", "features": ["content.code.annotate", "content.tooltips", "navigation.indexes", "navigation.sections", "navigation.footer", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow"], "search": "assets/javascripts/workers/search.f886a092.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script> <script src=assets/javascripts/bundle.cd18aaf1.min.js></script> <script src=javascripts/mathjax.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>