<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.550">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2023-11-30">

<title>RDM for NGS</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../img/logo.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">RDM for NGS</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../develop/01_RDM_intro.html"> 
<span class="menu-text">Course material</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../use_cases.html"> 
<span class="menu-text">Use cases</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../develop/practical_workshop.html"> 
<span class="menu-text">Practical material</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/hds-sandbox"> <i class="bi bi-github" role="img" aria-label="GitHub">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/company/ucph-heads/"> <i class="bi bi-linkedin" role="img" aria-label="LinkedIn">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#file-structure-and-naming-conventions" id="toc-file-structure-and-naming-conventions" class="nav-link active" data-scroll-target="#file-structure-and-naming-conventions">File structure and naming conventions</a>
  <ul class="collapse">
  <li><a href="#folder-organization" id="toc-folder-organization" class="nav-link" data-scroll-target="#folder-organization">Folder organization</a></li>
  <li><a href="#assay-folder" id="toc-assay-folder" class="nav-link" data-scroll-target="#assay-folder">Assay folder</a>
  <ul class="collapse">
  <li><a href="#assay-id-code-names" id="toc-assay-id-code-names" class="nav-link" data-scroll-target="#assay-id-code-names">Assay ID code names</a></li>
  <li><a href="#assay-folder-structure" id="toc-assay-folder-structure" class="nav-link" data-scroll-target="#assay-folder-structure">Assay folder structure</a></li>
  </ul></li>
  <li><a href="#project-folder" id="toc-project-folder" class="nav-link" data-scroll-target="#project-folder">Project folder</a>
  <ul class="collapse">
  <li><a href="#project-folder-structure" id="toc-project-folder-structure" class="nav-link" data-scroll-target="#project-folder-structure">Project folder structure</a></li>
  </ul></li>
  <li><a href="#template-engine" id="toc-template-engine" class="nav-link" data-scroll-target="#template-engine">Template engine</a>
  <ul class="collapse">
  <li><a href="#quick-tutorial-on-cookiecutter" id="toc-quick-tutorial-on-cookiecutter" class="nav-link" data-scroll-target="#quick-tutorial-on-cookiecutter">Quick tutorial on cookiecutter</a></li>
  </ul></li>
  <li><a href="#genomic-resources-folder" id="toc-genomic-resources-folder" class="nav-link" data-scroll-target="#genomic-resources-folder">Genomic resources folder</a>
  <ul class="collapse">
  <li><a href="#refgenie" id="toc-refgenie" class="nav-link" data-scroll-target="#refgenie">RefGenie</a></li>
  <li><a href="#manual-download" id="toc-manual-download" class="nav-link" data-scroll-target="#manual-download">Manual download</a></li>
  </ul></li>
  <li><a href="#naming-conventions" id="toc-naming-conventions" class="nav-link" data-scroll-target="#naming-conventions">Naming conventions</a>
  <ul class="collapse">
  <li><a href="#general-tips" id="toc-general-tips" class="nav-link" data-scroll-target="#general-tips">General tips</a></li>
  <li><a href="#suggestions-for-ngs-data" id="toc-suggestions-for-ngs-data" class="nav-link" data-scroll-target="#suggestions-for-ngs-data">Suggestions for NGS data</a></li>
  <li><a href="#print-a-table" id="toc-print-a-table" class="nav-link" data-scroll-target="#print-a-table">Print a table</a></li>
  </ul></li>
  <li><a href="#wrap-up" id="toc-wrap-up" class="nav-link" data-scroll-target="#wrap-up">Wrap up</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Best Practices for Data Storage</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 30, 2023</p>
    </div>
  </div>
  
    <div>
    <div class="quarto-title-meta-heading">Modified</div>
    <div class="quarto-title-meta-contents">
      <p class="date-modified">March 8, 2024</p>
    </div>
  </div>
    
  </div>
  


</header>


<!-- 
tags:
    - File structure
    - File organization
    - Naming conventions
    - Folder templates
-->
<div class="callout callout-style-default callout-note callout-titled" title="Course Overview">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Course Overview
</div>
</div>
<div class="callout-body-container callout-body">
<p>‚è∞ <strong>Time Estimation:</strong> X minutes<br>
üí¨ <strong>Learning Objectives:</strong></p>
<pre><code>1. Learn how to organize your NGS data.
2. Separate your data into projects and assays.
3. Manage genomic resources.
4. Learn general rules for naming files and folders.
5. Create your own rules for naming specific results and figures.</code></pre>
</div>
</div>
<section id="file-structure-and-naming-conventions" class="level1">
<h1>File structure and naming conventions</h1>
<p>We have learn so far how to adhere to FAIR and Open Science standards. You may have noticed that they mostly cover information regarding sharing your data after your project is finished or your data is published. However, how can you <strong>effectively manage your data while you work with it</strong>? How should I organize my data folders, raw data, processed data, data analysis scripts, figures and results? You may not worry too much about it since you will be working on your data all the time, so you will know where everything is. However, projects tend to last long, and the data, its analysis and results, accumulate greatly over time. Not having any sort of structure will incur in many problems in the future, specially once the project is finished or some colleague or collaborator want to access the data. How will they be able to find X or Y table? Will you be able to remember where a figure came from or which script created a csv file? By not applying an efficient and clear system, you are inviting chaos to your doorstep!</p>
<p>On the other hand, applying a consistent file structure and naming conventions to your files will help you to efficiently manage your data. Consider the following practices:</p>
<ul>
<li><strong>Folder structure</strong>: Aim to establish a logical and intuitive folder structure that reflects the organization of research projects and experimental data. Use descriptive folder names to make it easy to locate and access specific data files.
<ul>
<li><strong>Subfolders</strong>: Use subfolders to further categorize data based on their contents, such as code notebooks, results, reports, etc. This helps to keep data organized and facilitates quick retrieval.</li>
</ul></li>
<li><strong>File naming conventions</strong>: implement a standardized file naming convention to ensure consistency and clarity. Use descriptive names that include relevant information, such as type of plots, results tables, etc.</li>
</ul>
<p>In this lesson we will see a practical example on how you could organize your own files and folders.</p>
<section id="folder-organization" class="level2">
<h2 class="anchored" data-anchor-id="folder-organization">Folder organization</h2>
<p>You will probably want to divide your NGS data into three different types of folders:</p>
<ol type="1">
<li><strong>Assay folders</strong>: This folder contains the <strong>raw and processed NGS datasets</strong>, as well as the <strong>pipeline/workflow</strong> used to generate the processed data, provenance of the raw data and quality control reports of the data. This data should be <strong>locked and read-only</strong> to prevent unwanted modifications.</li>
<li><strong>Project folders</strong>: This folder contains <strong>all the necessary files for a specific research project</strong>. A project may use several assays or results from other projects. The assay data should not be copied or duplicated, but linked from the original source.</li>
<li><strong>Genomic resources folders</strong>: This folder contains <strong>common genomic resources</strong>, such as genome references (fasta files) and annotations (gtf files) for different species, as well as indexes for different alignment algorithms. This data should also be <strong>locked and read-only</strong> to prevent unwanted modifications.</li>
</ol>
<p>This will help you to keep your data tidied up, specially if you are working on a big lab where assays may be used for different purposes and different people!</p>
</section>
<section id="assay-folder" class="level2">
<h2 class="anchored" data-anchor-id="assay-folder">Assay folder</h2>
<p>For each NGS experiment there should be an <code>Assay</code> folder that will contain all experimental datasets, that is, an <code>Assay</code> (raw files and pipeline processed files). Raw files should not be modified at all, but you should probably lock modifications to the final results once you are done with preprocessing the data. This will help you prevent unwanted modifications to the data. Each <code>Assay</code> subfolder should be named in a way that it is unique, easily readable, distinguishable and understood at a glance. For example, you could name an NGS assay using an acronym for the type of NGS assay (RNAseq, ChIPseq, ATACseq), a keyword that represents a unique descriptive element of that assay, and the date. Like this:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>Assay-ID<span class="op">&gt;</span>_<span class="op">&lt;</span>keyword<span class="op">&gt;</span>_YYYYMMDD</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>For example <code>CHIP_Oct4_20230101</code> is a ChIPseq assay made on 1st January 2023 with the keyword Oct4, so it is easily identifiable by eye.</p>
<section id="assay-id-code-names" class="level3">
<h3 class="anchored" data-anchor-id="assay-id-code-names">Assay ID code names</h3>
<p>Below is a example list of different Assay-ID code names. You are welcome to use it and expand it as you wish!</p>
<ul>
<li><code>CHIP</code>: ChIP-seq</li>
<li><code>RNA</code>: RNA-seq</li>
<li><code>ATAC</code>: ATAC-seq</li>
<li><code>SCR</code>: scRNA-seq</li>
<li><code>PROT</code>: Mass Spectrometry Assay</li>
<li><code>CAT</code>: Cut&amp;Tag</li>
<li><code>CAR</code>: Cut&amp;Run</li>
<li><code>RIME</code>: Rapid Immunoprecipitation Mass spectrometry of Endogenous proteins</li>
<li>‚Ä¶</li>
</ul>
</section>
<section id="assay-folder-structure" class="level3">
<h3 class="anchored" data-anchor-id="assay-folder-structure">Assay folder structure</h3>
<p>Next, let‚Äôs take a look at a possible folder structure and what kind of files you can find there.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">CHIP_Oct4_20230101/</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> README.md</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> metadata.yml</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> pipeline.md</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> processed</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îî‚îÄ‚îÄ</span> raw</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>   <span class="ex">‚îú‚îÄ‚îÄ</span> .fastq.gz</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>   <span class="ex">‚îî‚îÄ‚îÄ</span> samplesheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><strong>README.md</strong>: Long description of the assay in markdown format. It should contain provenance of the raw NGS data (samples, laboratory protocols used, aim of the assay, etc)</li>
<li><strong>metadata.yml</strong>: metadata file for the assay describing different keys and important information regarding that assay (<a href="./07_metadata.md">see this lesson</a>).</li>
<li><strong>pipeline.md</strong>: description of the pipeline used to process raw data, as well as the commands used to run the pipeline.</li>
<li><strong>processed</strong>: folder with results of the preprocessing pipeline. Contents depend on the pipeline used.</li>
<li><strong>raw</strong>: folder with the raw data.
<ul>
<li><em>.fastq.gz</em>:In the case of NGS assays, there should be fastq files.</li>
<li><em>samplesheet.csv</em>: file that contains metadata information for the samples. This file is used to run the nf-core pipelines. You can also add extra columns with info regarding the experimental variables and batches so it can be used for downstream analysis as well.</li>
</ul></li>
</ul>
<p>This folder structure is simple and straight forward, and files will be exactly where you expect them to be! By including a description and metadata file, you will always be able to understand what this experiment was and how it was generated. This information will be <strong>specially useful once you need to submit your data for archiving or write a manuscript</strong>. The <strong>metadata</strong> file will be really handy if you want to <strong>collect information on all your NGS datasets and create a database</strong> (<a href="./08_database.md">see this lesson</a>)!</p>
<p>Note that the processed folder is not expanded upon. This folder will be very dependant on the workflows/pipelines that you use. We strongly recommend that you use reproducible and community curated pipelines, such as the ones developed by the <a href="https://nf-co.re/">nf-core community</a>, which have a through documentation on the results they produce. For example, imagine you have run an RNAseq experiment and processed the data using the <a href="https://nf-co.re/rnaseq/3.12.0">nf-core:rnaseq pipeline</a> with the pseudoalignment and quantification option:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">processed/</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> fastqc/</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> multiqc/</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> pipeline_info/</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> salmon/</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îî‚îÄ‚îÄ</span> trimgalore/</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><strong>fastqc</strong>: Quality Control results of the raw fastq files.</li>
<li><strong>multiqc</strong>: Full compilation of the Quality Control checks for all your samples for the entire pipeline.</li>
<li><strong>pipeline_info</strong>: Information and logs about the pipeline used for each of the steps of the workflow.</li>
<li><strong>salmon</strong>: Pseudoalignment and quantification results of the salmon algorithm.</li>
<li><strong>trimgalore</strong>: Cleaned fastq files and Quality Control results of the cleaned files.</li>
</ul>
<p>By using a standardized pipeline, you should be able to always find the preprocessing results where you expect them to be! No need to come up with your own file organization (and document it) or worry that your colleagues (or your future self) won‚Äôt be able to find the data.</p>
</section>
</section>
<section id="project-folder" class="level2">
<h2 class="anchored" data-anchor-id="project-folder">Project folder</h2>
<p>On the other hand, we have the other type of folder called <code>Projects</code>. In this folder you will save a <strong>subfolder for each project</strong> that you (or your lab) works on. Each <code>Project</code> subfolder will contain project information and all the data analysis notebooks and scripts used in that project.</p>
<p>Projects and Assays are separated from each other because a project may use one or more assays to answer a scientific question, and assays may be reused several times in different projects. This could be, for example, all the data analysis related to a publication (a RNAseq and a ChIPseq experiment), or a comparison between a previous ATACseq experiment (which was used for a older project) with a new laboratory protocol.</p>
<p>As like for an Assay folder, the Project folder should be named in a way that it is unique, easily readable, distinguishable and understood at a glance. For example, you could name it after the main author initials, a keyword that represents a unique descriptive element of that assay, and the date:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>author_initials<span class="op">&gt;</span>_<span class="op">&lt;</span>keyword<span class="op">&gt;</span>_YYYYMMDD</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>For example, <code>JARH_Oct4_20230101</code>, is a project about the gene Oct4 owned by Jose Alejandro Romero Herrera, created on the 1st of January of 2023.</p>
<section id="project-folder-structure" class="level3">
<h3 class="anchored" data-anchor-id="project-folder-structure">Project folder structure</h3>
<p>Next, let‚Äôs take a look at a possible folder structure and what kind of files you can find there.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>author_initials<span class="op">&gt;</span>_<span class="op">&lt;</span>keyword<span class="op">&gt;</span>_YYYYMMDD</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> data</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îî‚îÄ‚îÄ <span class="op">&lt;</span>Assay-ID<span class="op">&gt;</span>_<span class="op">&lt;</span>keyword<span class="op">&gt;</span>_YYYYMMDD/</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> documents</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îî‚îÄ‚îÄ Non-sensitive_NGS_research_project_template.docx</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> notebooks</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îî‚îÄ‚îÄ 01_data_analysis.rmd</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> README.md</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> reports</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îú‚îÄ‚îÄ figures</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îÇ  ‚îî‚îÄ‚îÄ 01_data_analysis/</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îÇ   ‚îî‚îÄ‚îÄ heatmap_sampleCor_20230102.png</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îî‚îÄ‚îÄ 01_data_analysis.html</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> requirements.txt</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> results</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îî‚îÄ‚îÄ 01_data_analysis/</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>      ‚îî‚îÄ‚îÄ DEA_treat-control_LFC1_p01.tsv</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> scripts</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îî‚îÄ‚îÄ</span> metadata.yml</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><strong>data</strong>: folder that contains symlinks or shortcuts to where the data is, avoiding copying and modification of original files.</li>
<li><strong>documents</strong>: folder containing word documents, slides or pdfs related to the project, such as explanations of the data or project, papers, etc. It also contains your <a href="./05_DMP.md">Data Management Plan</a>.
<ul>
<li><em>Non-sensitive_NGS_research_project_template.docx</em>. This is a pre-filled Data Management Plan based on the Horizon Europe guidelines.</li>
</ul></li>
<li><strong>notebooks</strong>: folder containing Jupyter, R markdown or Quarto notebooks with the actual data analysis.</li>
<li><strong>README.md</strong>: detailed description of the project in markdown format.</li>
<li><strong>reports</strong>: notebooks rendered as html/docx/pdf versions, ideal for sharing with colleagues and also as a formal report of the data analysis procedure.
<ul>
<li><em>figures</em>: figures produced upon rendering notebooks. The figures will be saved under a subfolder named after the notebook that created them. This is for provenance purposes so we know which notebook created which figures.</li>
</ul></li>
<li><strong>requirements.txt</strong>: file explaining what software and libraries/packages and their versions are necessary to reproduce the code.</li>
<li><strong>results</strong>: results from the data analysis, such as tables with differentially expressed genes, enrichment results, etc.</li>
<li><strong>scripts</strong>: folder containing helper scripts needed to run data analysis or reproduce the work of the folder</li>
<li><strong>description.yml</strong>: short description of the project.</li>
<li><strong>metadata.yml</strong>: metadata file for the assay describing different keys (<a href="./07_metadata.md">see this lesson</a>).</li>
</ul>
<div class="callout callout-style-default callout-note callout-titled" title="Note on the **data** folder">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note on the <strong>data</strong> folder
</div>
</div>
<div class="callout-body-container callout-body">
<pre><code>What do we mean with shortcuts to the data is? If you have divided your Assays and Projects as suggested, you do not really need to copy or duplicate your data again, simply make a shortcut to the original folder, and you will have access to the data as if it is in the project folder! You can use this command in Linux or MacOS to create a folder softlink:

```{.bash}
ln -s path/to/assays/Assays/&lt;ASSAY_ID&gt; /path/to/projects/Projects/&lt;PROJECT_ID&gt;/data/
```</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled" title="Note on the **notebooks** folder">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note on the <strong>notebooks</strong> folder
</div>
</div>
<div class="callout-body-container callout-body">
<pre><code>Using annotated notebooks is ideal for reproducibility and readability purposes. Notebooks should be labeled numerically in order they were created, or the order of the data analysis steps, if they are related to each other e.g. `00_preprocessing.rmd` - `01_Differential_Expression_Analysis.rmd` - `02_Functional_Analysis.rmd`</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled" title="Note on the **results** folder">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note on the <strong>results</strong> folder
</div>
</div>
<div class="callout-body-container callout-body">
<pre><code>Results from your code notebooks should be saved under this folder. Create a subfolder named after the notebook that created them, so you can always identify which notebook created which results!</code></pre>
</div>
</div>
<!-- **Note**: maybe we can make an environment folder or put inside the scripts folder a Dockerfile that should give you an environment where you can reproduce the results of the folder? -->
</section>
</section>
<section id="template-engine" class="level2">
<h2 class="anchored" data-anchor-id="template-engine">Template engine</h2>
<p>You may be thinking right now that doing this kind of setup every time you need to create a new folder would be very tedious, and you would be right! Fortunately, it is very easy to create a folder template using <a href="https://github.com/cookiecutter/cookiecutter">cookiecutter</a>. Cookiecutter is a command-line utility that creates projects from cookiecutters (that is, a template), e.g.&nbsp;creating a Python package project from a Python package project template. Here you can find an example of a cookiecutter folder template directed to <a href="https://github.com/brickmanlab/ngs-template">NGS data</a>, where we have applied the structures explained in the previous sections. You are very welcome to adapt it or modify it to your needs!</p>
<p>Now, using only cookiecutter might have some issues. If you update the template, it might be really hard to maintain older folders. By using another tool, <a href="https://github.com/cruft/cruft">cruft</a>, when generating assay and project folders, it allows the user to validate and syncronize old templates with the latest version.</p>
<section id="quick-tutorial-on-cookiecutter" class="level3">
<h3 class="anchored" data-anchor-id="quick-tutorial-on-cookiecutter">Quick tutorial on cookiecutter</h3>
<p>Creating a Cookiecutter template from scratch involves defining a folder structure, creating a <code>cookiecutter.json</code> file, and specifying the placeholders (keywords) that will be replaced during project generation. Let‚Äôs walk through the process step by step:</p>
<section id="step-1-create-a-folder-template" class="level4">
<h4 class="anchored" data-anchor-id="step-1-create-a-folder-template">Step 1: Create a Folder Template</h4>
<p>Start by creating a folder with the structure you want for your template. For example, let‚Äôs create a simple Python project template:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">my_template/</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span><span class="ex">--</span> {{cookiecutter.project_name}}</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span>   <span class="kw">|</span><span class="ex">--</span> main.py</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span><span class="ex">--</span> tests</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span>   <span class="kw">|</span><span class="ex">--</span> test_{{cookiecutter.project_name}}.py</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span><span class="ex">--</span> README.md</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>In this example, <code>{cookiecutter.project_name}</code> is a placeholder that will be replaced with the actual project name when the template is used.</p>
</section>
<section id="step-2-create-cookiecutter.json" class="level4">
<h4 class="anchored" data-anchor-id="step-2-create-cookiecutter.json">Step 2: Create <code>cookiecutter.json</code></h4>
<p>In the root of your template folder, create a file named <code>cookiecutter.json</code>. This file will define the variables (keywords) that users will be prompted to fill in. For our Python project template, it might look like this:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">{</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="dt">"project_name"</span><span class="fu">:</span> <span class="st">"MyProject"</span><span class="fu">,</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="dt">"author_name"</span><span class="fu">:</span> <span class="st">"Your Name"</span><span class="fu">,</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="dt">"description"</span><span class="fu">:</span> <span class="st">"A short description of your project"</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>These are the questions users will be asked when generating a project based on your template. The values provided here will be used to replace the corresponding placeholders in the template files.</p>
<p>In addition to replacing placeholders in file and directory names, Cookiecutter can also automatically fill in information within the contents of text files. This can be useful for providing default configurations or templates for code files. Let‚Äôs extend our previous example to include a placeholder inside a text file:</p>
<p>First, modify the <code>my_template/main.py</code> file to include a placeholder inside its contents:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># main.py</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> hello():</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Hello, </span><span class="sc">{{</span><span class="st">cookiecutter.project_name</span><span class="sc">}}</span><span class="st">!"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Now, the <code>{cookiecutter.project_name}</code> placeholder is inside the <code>main.py</code> file. When you run Cookiecutter, it will automatically replace the placeholders not only in file and directory names but also within the contents of text files. After running Cookiecutter, your generated <code>main.py</code> file might look like this:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># main.py</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> hello():</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="st">"Hello, MyProject!"</span>)  <span class="co"># Assuming "MyProject" was entered as the project_name</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="step-3-use-cookiecutter" class="level4">
<h4 class="anchored" data-anchor-id="step-3-use-cookiecutter">Step 3: Use Cookiecutter</h4>
<p>Now that your template is set up, you can use Cookiecutter to generate a project based on it. Open a terminal and run:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cookiecutter</span> path/to/your/template</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Cookiecutter will prompt you to fill in the values for <code>project_name</code>, <code>author_name</code>, and <code>description</code>. After you provide these values, Cookiecutter will replace the placeholders in your template files with the entered values.</p>
</section>
<section id="step-4-explore-the-generated-project" class="level4">
<h4 class="anchored" data-anchor-id="step-4-explore-the-generated-project">Step 4: Explore the Generated Project</h4>
<p>Once the generation process is complete, navigate to the directory where Cookiecutter created the new project. You will see a project structure with the placeholders replaced by the values you provided.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Exercise: Create your own template‚Äù
</div>
</div>
<div class="callout-body-container callout-body">
<p>Using cookiecutter, create your own templates for your folders. You do not need to copy exactly our suggestions, adjust your template to your own needs! In order to create your cookiecutter template, your will need to install python, <a href="https://cookiecutter.readthedocs.io/en/stable/installation.html">cookiecutter</a>, Git and a GitHub account. If you do not have Git and a GitHub account, we suggest you do one as soon as possible. We will take a deeper look at Git and GitHub in the <a href="./09_version_control.md">version control lesson</a>.</p>
<p>We have prepared already two simple cookiecutter templates in GitHub repositories.</p>
<pre><code>**Assay**

1. First, fork our `https://github.com/hds-sandbox/assay-template` from the GitHub page into your own account/organization.
![fork_repo_example](./images/fork_repo.png)
2. Then, use `git clone &lt;your URL to the template&gt;` to put it in your computer.
3. Modify the contents of the repository so that it matches the **Assay** example above. You are welcome to do changes as you please!
4. Modify the `cookiecutter.json` file so that it will include the **Assay** name template
5. Git add, commit and push your changes
6. Test your folder by using `cookiecutter &lt;URL to your GitHub repository for "assay-template&gt;`

**Project**

7. First, fork our `https://github.com/hds-sandbox/project-template` from the GitHub page into your own account/organization.
![fork_repo_example](./images/fork_repo_project.png)
8. Then, use `git clone &lt;your URL to the template&gt;` to put it in your computer.
9.  Modify the contents of the repository so that it matches the **Project** example above. You are welcome to do changes as you please!
10. Modify the `cookiecutter.json` file so that it will include the **Project** name template
11. Git add, commit and push your changes
12. Test your folder by using `cookiecutter &lt;URL to your GitHub repository for "project-template&gt;`</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="genomic-resources-folder" class="level2">
<h2 class="anchored" data-anchor-id="genomic-resources-folder">Genomic resources folder</h2>
<p>Preprocessing NGS data usually requires different genomic resources in order to align and and annotate fastq files. First and most importantly, you will need reference genomes (human and mouse are the most common ones) in FASTA format. In addition, aligner tools such as STAR, Bowtie, etc., require indexed fasta files to perform alignment of reads. Moreover, GTF or GFF files are necessary in order to quantify reads into genomic regions (such as genes or promoters). Nonetheless, these genomic resources are often updated and are released periodically under different versions from different sources. In order to make your data reproducible, you will need to control and manage these genomic resources and specify their versions and sources. For example, the latest mouse reference genome is GRCm39, but may studies still align their reads to the version GRCm38.</p>
<p>How do you keep track of your resources? You could, again, set up a folder structure that hosts reference genomes, annotations and indexes per species and per version, or use a reference genome manager such as <a href="http://refgenie.databio.org/en/latest/">refgenie</a>. We will take a look at both options.</p>
<section id="refgenie" class="level3">
<h3 class="anchored" data-anchor-id="refgenie">RefGenie</h3>
<p><a href="http://refgenie.databio.org/en/latest/">Refgenie</a> manages storage, access, and transfer of reference genome resources. It provides command-line and Python interfaces to download pre-built reference genome ‚Äúassets‚Äù, like indexes used by bioinformatics tools. It can also build assets for custom genome assemblies. Refgenie provides programmatic access to a standard genome folder structure, so software can swap from one genome to another.</p>
<p>Check <a href="http://refgenie.databio.org/en/latest/tutorial/">their tutorial</a> if you want to start managing your genomic resources with Refgenie!</p>
</section>
<section id="manual-download" class="level3">
<h3 class="anchored" data-anchor-id="manual-download">Manual download</h3>
<p>If you want to keep manual track and manage yourself and not depend on a third party tool, you can always download the genomic resources directly from the source. We will show you how in this section:</p>
<ol type="1">
<li>Create a new folder, called <code>genomic_resources</code>.</li>
<li>Inside this folder, separate each species with a subfolder.</li>
<li>Inside the species subfolder, separate it by the resource version</li>
<li>Inside the version subfolder, separate genomic sequences (FASTA) and annotations (GTF/GFF) from different alignment indexes</li>
</ol>
<p>In the example below, we are downloading human (version GRCh38) and mouse (version GRCm39) genomic resources from the <a href="ftp://ftp.ensembl.org">ensembl FTP server</a></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">genomic_resources/</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> homo_sapiens/</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îî‚îÄ‚îÄ GRCh38/</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>     ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh38.109.gtf.gz</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>     ‚îú‚îÄ‚îÄ Homo_sapiens.GRCh38.109.dna_sm.primary_assembly.fa.gz</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>     ‚îî‚îÄ‚îÄ indexes/</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>        ‚îú‚îÄ‚îÄ salmon/</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>        ‚îî‚îÄ‚îÄ STAR/</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> mus_musculus/</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>  ‚îî‚îÄ‚îÄ GRCm39/</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>     ‚îú‚îÄ‚îÄ Mus_musculus.GRCm39.109.gtf.gz</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>     ‚îú‚îÄ‚îÄ Mus_musculus.GRCm39.109.dna_sm.primary_assembly.fa.gz</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>     ‚îî‚îÄ‚îÄ indexes/</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>        ‚îú‚îÄ‚îÄ salmon/</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îÇ</span>        ‚îî‚îÄ‚îÄ STAR/</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îú‚îÄ‚îÄ</span> ref_genomes.sh</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="ex">‚îî‚îÄ‚îÄ</span> create_indexes.sh</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>As you can see, we have also keep here a bash script that downloads and sets up the folders. The contents of the bash script should also indicate a date of when was the data downloaded. For example:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Download human (GRCh38) and mouse (GRCm39) genome and annotations</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Release 109</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 2023-07-10</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Go to home</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Create homo_sapiens version folder and go to that folder</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> genomic_resources/homo_sapiens/GRCh38</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span>  genomic_resources/homo_sapiens/GRCh38</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Download FASTA file and GTF file from the ensembl release 109</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-L</span> ftp://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa.gz</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-L</span> ftp://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Go to home</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Create mus_musculus version folder and go to that folder</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> genomic_resources/mouse/GRCm39</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span>  genomic_resources/mouse/GRCm39</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Download FASTA file and GTF file from the ensembl release 109</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-L</span> ftp://ftp.ensembl.org/pub/release-109/fasta/mus_musculus/dna/Mus_musculus.GRCm39.dna_sm.primary_assembly.fa.gz</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-L</span> https://ftp.ensembl.org/pub/release-109/gtf/mus_musculus/Mus_musculus.GRCm39.109.gtf.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>After that is all set up, you can run another bash script that will create a index folder inside each version subfolder and generate indexes for different alignment tools!</p>
</section>
</section>
<section id="naming-conventions" class="level2">
<h2 class="anchored" data-anchor-id="naming-conventions">Naming conventions</h2>
<p>Using consistent naming conventions is important in scientific research as it helps with the organization and retrieval of data or results. By adopting standardized naming conventions, researchers ensure that files, experiments, or data sets are labeled in a clear, logical manner. This makes it easier to locate and compare similar types of data or results, even when dealing with large datasets or multiple experiments. For instance, in genomics, employing uniform naming conventions for files related to specific experiments or samples allows for swift identification and comparison of relevant data, streamlining the research process and contributing to the reproducibility of findings. This practice promotes efficiency, collaboration, and the integrity of scientific work.</p>
<section id="general-tips" class="level3">
<h3 class="anchored" data-anchor-id="general-tips">General tips</h3>
<p>Below you will find a small list of general tips to follow when you name a folder or a file:</p>
<ul>
<li>Use only alphanumeric characters to write a word: a to z and 0 to 9</li>
<li>Avoid special characters: ~!@#$%^&amp;*()`<span></span>‚Äú|</li>
<li>Date format: use <code>YYYYMMDD</code> format. For example: 20230101.</li>
<li>Authors: use initials. For example: JARH</li>
<li><strong>Don‚Äôt use of spaces</strong>! Computers get very confused when you need to point a path to a file and it contains spaces! Instead:
<ul>
<li>Separate field sections are separated by underscores <code>_</code>.</li>
<li>Words in each section are written in <a href="https://en.wikipedia.org/wiki/Camel_case">camelCase</a>. It would look then like this: <code>field1_word1Word2.txt</code>. For example: <code>heatmap_sampleCor_20230101.png</code>. The first field indicates what this file is, i.e., a heatmap. Second field is what is being plotted, i.e, sample correlations; since the field contains two words, they are written in camelCase. The third field is the date of when the image was created.</li>
</ul></li>
<li>Use as short fields as possible. You can try to use understandable abbreviations, like LFC for LogFoldChange, Cor for correlations, Dist for distances, etc.</li>
<li>Avoid long names as much as you can, be concise!</li>
<li>Avoid creating many sublevels of folders.</li>
<li>Write down your naming convention pattern and document it in the README file</li>
<li>When using a sequential numbering system, use leading zeros to make sure files sort in sequential order. Using <code>01</code> instead of just <code>1</code> if your sequence only goes up to <code>99</code>.</li>
<li>Versions should be used as the last element, and use at least two digits with a leading 0 (e.g.&nbsp;v01, v02)</li>
</ul>
</section>
<section id="suggestions-for-ngs-data" class="level3">
<h3 class="anchored" data-anchor-id="suggestions-for-ngs-data">Suggestions for NGS data</h3>
<p>More info on naming conventions for different types of files and analysis is in development.</p>
</section>
<section id="print-a-table" class="level3">
<h3 class="anchored" data-anchor-id="print-a-table">Print a table</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<div id="jxapfhdpbo" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#jxapfhdpbo table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#jxapfhdpbo thead, #jxapfhdpbo tbody, #jxapfhdpbo tfoot, #jxapfhdpbo tr, #jxapfhdpbo td, #jxapfhdpbo th {
  border-style: none;
}

#jxapfhdpbo p {
  margin: 0;
  padding: 0;
}

#jxapfhdpbo .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 11px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#jxapfhdpbo .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#jxapfhdpbo .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#jxapfhdpbo .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#jxapfhdpbo .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jxapfhdpbo .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jxapfhdpbo .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jxapfhdpbo .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#jxapfhdpbo .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#jxapfhdpbo .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#jxapfhdpbo .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#jxapfhdpbo .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#jxapfhdpbo .gt_spanner_row {
  border-bottom-style: hidden;
}

#jxapfhdpbo .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#jxapfhdpbo .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#jxapfhdpbo .gt_from_md > :first-child {
  margin-top: 0;
}

#jxapfhdpbo .gt_from_md > :last-child {
  margin-bottom: 0;
}

#jxapfhdpbo .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#jxapfhdpbo .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#jxapfhdpbo .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#jxapfhdpbo .gt_row_group_first td {
  border-top-width: 2px;
}

#jxapfhdpbo .gt_row_group_first th {
  border-top-width: 2px;
}

#jxapfhdpbo .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jxapfhdpbo .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#jxapfhdpbo .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#jxapfhdpbo .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jxapfhdpbo .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jxapfhdpbo .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#jxapfhdpbo .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#jxapfhdpbo .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#jxapfhdpbo .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jxapfhdpbo .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jxapfhdpbo .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jxapfhdpbo .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jxapfhdpbo .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jxapfhdpbo .gt_left {
  text-align: left;
}

#jxapfhdpbo .gt_center {
  text-align: center;
}

#jxapfhdpbo .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#jxapfhdpbo .gt_font_normal {
  font-weight: normal;
}

#jxapfhdpbo .gt_font_bold {
  font-weight: bold;
}

#jxapfhdpbo .gt_font_italic {
  font-style: italic;
}

#jxapfhdpbo .gt_super {
  font-size: 65%;
}

#jxapfhdpbo .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#jxapfhdpbo .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#jxapfhdpbo .gt_indent_1 {
  text-indent: 5px;
}

#jxapfhdpbo .gt_indent_2 {
  text-indent: 10px;
}

#jxapfhdpbo .gt_indent_3 {
  text-indent: 15px;
}

#jxapfhdpbo .gt_indent_4 {
  text-indent: 20px;
}

#jxapfhdpbo .gt_indent_5 {
  text-indent: 25px;
}
</style>

<table class="gt_table table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="header gt_col_headings">
<th id="name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">name</th>
<th id="description" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">description</th>
<th id="naming_convention" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">naming_convention</th>
<th id="file format" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">file format</th>
<th id="example" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">example</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="name">.fastq</td>
<td class="gt_row gt_left" headers="description">raw sequencing reads</td>
<td class="gt_row gt_left" headers="naming_convention">nan</td>
<td class="gt_row gt_left" headers="file format">nan</td>
<td class="gt_row gt_left" headers="example">sampleID_run_read1.fastq</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">.fastqc</td>
<td class="gt_row gt_left" headers="description">quality control from fastqc</td>
<td class="gt_row gt_left" headers="naming_convention">nan</td>
<td class="gt_row gt_left" headers="file format">nan</td>
<td class="gt_row gt_left" headers="example">sampleID_run_read1.fastqc</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">.bam</td>
<td class="gt_row gt_left" headers="description">aligned reads</td>
<td class="gt_row gt_left" headers="naming_convention">nan</td>
<td class="gt_row gt_left" headers="file format">nan</td>
<td class="gt_row gt_left" headers="example">sampleID_run_read1.bam</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">GTF</td>
<td class="gt_row gt_left" headers="description">sequence annotation</td>
<td class="gt_row gt_left" headers="naming_convention">nan</td>
<td class="gt_row gt_left" headers="file format">nan</td>
<td class="gt_row gt_left" headers="example">one of https://www.gencodegenes.org/</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">GFF</td>
<td class="gt_row gt_left" headers="description">sequence annotation</td>
<td class="gt_row gt_left" headers="naming_convention">nan</td>
<td class="gt_row gt_left" headers="file format">nan</td>
<td class="gt_row gt_left" headers="example">one of https://www.gencodegenes.org/</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">.bed</td>
<td class="gt_row gt_left" headers="description">genome locations</td>
<td class="gt_row gt_left" headers="naming_convention">nan</td>
<td class="gt_row gt_left" headers="file format">nan</td>
<td class="gt_row gt_left" headers="example">nan</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">.bigwig</td>
<td class="gt_row gt_left" headers="description">genome coverage</td>
<td class="gt_row gt_left" headers="naming_convention">nan</td>
<td class="gt_row gt_left" headers="file format">nan</td>
<td class="gt_row gt_left" headers="example">nan</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">.fasta</td>
<td class="gt_row gt_left" headers="description">sequence data (nucleotide/aminoacid)</td>
<td class="gt_row gt_left" headers="naming_convention">nan</td>
<td class="gt_row gt_left" headers="file format">nan</td>
<td class="gt_row gt_left" headers="example">one of https://www.gencodegenes.org/</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">Multiqc report</td>
<td class="gt_row gt_left" headers="description">QC aggregated report</td>
<td class="gt_row gt_left" headers="naming_convention">&lt;assayID\&gt;_YYYYMMDD.multiqc</td>
<td class="gt_row gt_left" headers="file format">multiqc</td>
<td class="gt_row gt_left" headers="example">RNA_20200101.multiqc</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">Count matrix</td>
<td class="gt_row gt_left" headers="description">final count matrix</td>
<td class="gt_row gt_left" headers="naming_convention">&lt;assayID\&gt;_cm_aligner_YYYYMMDD.tsv</td>
<td class="gt_row gt_left" headers="file format">tsv</td>
<td class="gt_row gt_left" headers="example">RNA_cm_salmon_20200101.tsv</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">DEA</td>
<td class="gt_row gt_left" headers="description">differential expression analysis results</td>
<td class="gt_row gt_left" headers="naming_convention">DEA_&lt;condition1-condition2\&gt;_LFC&lt;absolute_threshold\&gt;_p&lt;pvalue decimals\&gt;_YYYYMMDD.tsv</td>
<td class="gt_row gt_left" headers="file format">tsv</td>
<td class="gt_row gt_left" headers="example">DEA_treat-untreat_LFC1_p01_20200101.tsv</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">DBA</td>
<td class="gt_row gt_left" headers="description">differential binding analysis results</td>
<td class="gt_row gt_left" headers="naming_convention">DBA_&lt;condition1-condition2\&gt;_LFC&lt;absolute_threshold\&gt;_p&lt;pvalue decimals\&gt;_YYYYMMDD.tsv</td>
<td class="gt_row gt_left" headers="file format">tsv</td>
<td class="gt_row gt_left" headers="example">DBA_treat-untreat_LFC1_p01_20200101.tsv</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">MAplot</td>
<td class="gt_row gt_left" headers="description">MA plot</td>
<td class="gt_row gt_left" headers="naming_convention">MAplot_&lt;condition1-condition2\&gt;_YYYYMMDD.jpeg</td>
<td class="gt_row gt_left" headers="file format">jpeg</td>
<td class="gt_row gt_left" headers="example">MAplot_treat-untreat_20200101.jpeg</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">Heatmap plot</td>
<td class="gt_row gt_left" headers="description">Heatmap plot of anything</td>
<td class="gt_row gt_left" headers="naming_convention">heatmap_&lt;type\&gt;_YYYYMMDD.jpeg</td>
<td class="gt_row gt_left" headers="file format">jpeg</td>
<td class="gt_row gt_left" headers="example">heatmap_sampleCor_20200101.jpeg</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">Volcano plot</td>
<td class="gt_row gt_left" headers="description">Volcano plot</td>
<td class="gt_row gt_left" headers="naming_convention">volcano_&lt;condition1-condition2\&gt;_YYYYMMDD.jpeg</td>
<td class="gt_row gt_left" headers="file format">jpeg</td>
<td class="gt_row gt_left" headers="example">volcano_treat-untreat_20200101.jpeg</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="name">Venn diagram</td>
<td class="gt_row gt_left" headers="description">Venn diagram</td>
<td class="gt_row gt_left" headers="naming_convention">venn_&lt;type\&gt;_YYYYMMDD.jpeg</td>
<td class="gt_row gt_left" headers="file format">jpeg</td>
<td class="gt_row gt_left" headers="example">venn_consensus_20200101.jpeg</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="name">Enrichment table</td>
<td class="gt_row gt_left" headers="description">Enrichment results</td>
<td class="gt_row gt_left" headers="naming_convention">nan</td>
<td class="gt_row gt_left" headers="file format">tsv</td>
<td class="gt_row gt_left" headers="example">nan</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</div>
<!--  {{ read_table('./assets/file_naming_convention.tsv') }} -->
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
‚ÄúExercise: Create your own naming conventions‚Äù
</div>
</div>
<div class="callout-body-container callout-body">
<pre><code>Think about the most common types of files and folders you will be working on, such as visualizations, results tables, processed files, etc. Then come up with a logical and clear way of naming those files using the tips suggested above. Remember to avoid making long and complicated names!</code></pre>
</div>
</div>
</section>
</section>
<section id="wrap-up" class="level2">
<h2 class="anchored" data-anchor-id="wrap-up">Wrap up</h2>
<p>In this lesson, we have learned some practical tips and examples about how to organize your data and bring some order to chaos! We have shown how to use <code>cookiecutter</code> as a template engine that can be used to create new <code>Assay</code> and <code>Project</code> folders without hassle. Hopefully you are now able to create your own templates and reuse them as much as you need. In the next lesson, we will look at what kind of metadata you should record for your <code>Assay</code> and <code>Project</code> folders, from general fields such as author or date, to more NGS specific like organism, cell type or tissue.</p>


</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-copyright"><h2 class="anchored quarto-appendix-heading">Copyright</h2><div class="quarto-appendix-contents"><div>CC-BY-SA 4.0 license</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "Óßã";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>